commit c48a6cdef0cf53d97977554a17f600951546583f
Author: olszomal <Malgorzata.Olszowka@stunnel.org>
Date:   Tue Nov 29 10:03:19 2022 +0100

    Legacy pkcs12 ciphers tests
    Use legacy PKCS#12 container with RC2-40-CBC private key and certificate encryption algorithm

--- a/cmake/CMakeTest.cmake
+++ b/cmake/CMakeTest.cmake
@@ -20,6 +20,7 @@
   DESTINATION "${CONF}"
 )
 
+set(legacy_p12 "-pkcs12" "${CERTS}/legacy.p12" "-readpass" "${CERTS}/password.txt")
 set(priv_p12 "-pkcs12" "${CERTS}/cert.p12" "-readpass" "${CERTS}/password.txt")
 set(priv_spc "-certs" "${CERTS}/cert.spc" "-key" "${CERTS}/key.pvk" "-pass" "passme")
 set(priv_der "-certs" "${CERTS}/cert.pem" "-key" "${CERTS}/key.der" "-pass" "passme")
@@ -61,7 +62,7 @@
 )
 set(extensions_4 "exe" "ex_" "msi" "cat")
 set(extensions_3 "exe" "ex_" "msi")
-set(files_4 "signed" "nested" "added")
+set(files_4 "legacy" "signed" "nested" "added")
 set(files_3 "removed" "attached_pem" "attached_der")
 set(sign_formats "pem" "der")
 set(pem_certs "cert" "expired" "revoked")
@@ -74,7 +75,16 @@
 
 foreach(ext ${extensions_4})
   # Signing time: May  1 00:00:00 2019 GMT
-  set(sign_${ext} )
+  add_test(
+    NAME legacy_${ext}
+    COMMAND osslsigncode "sign" ${sign_opt} ${legacy_p12}
+      "-in" "${FILES}/unsigned.${ext}" "-out" "${FILES}/legacy.${ext}"
+  )
+endforeach()
+
+foreach(ext ${extensions_4})
+  # Signing time: May  1 00:00:00 2019 GMT
+  set(sign_${ext})
   add_test(
     NAME signed_${ext}
     COMMAND osslsigncode "sign" ${sign_opt} ${priv_p12}
@@ -280,6 +290,7 @@
 endif()
 
 foreach(ext ${extensions_4})
+  set(OUTPUT_FILES ${OUTPUT_FILES} "${FILES}/legacy.${ext}")
   set(OUTPUT_FILES ${OUTPUT_FILES} "${FILES}/signed.${ext}")
   set(OUTPUT_FILES ${OUTPUT_FILES} "${FILES}/nested.${ext}")
   set(OUTPUT_FILES ${OUTPUT_FILES} "${FILES}/removed.${ext}")
--- a/tests/certs/CACert.pem
+++ b/tests/certs/CACert.pem
@@ -1,22 +1,22 @@
 -----BEGIN CERTIFICATE-----
-MIIDoTCCAomgAwIBAgIUOK8lwJ8A1Oqw8jDAb3TF06ve+PcwDQYJKoZIhvcNAQEL
+MIIDoTCCAomgAwIBAgIUD5zM9oBx4OIRM05wnRjq2Ap2dlowDQYJKoZIhvcNAQEL
 BQAwWDELMAkGA1UEBhMCUEwxFTATBgNVBAoMDG9zc2xzaWduY29kZTEgMB4GA1UE
 CwwXQ2VydGlmaWNhdGlvbiBBdXRob3JpdHkxEDAOBgNVBAMMB1Jvb3QgQ0EwHhcN
 MTcwMTAxMDAwMDAwWhcNMjYxMTEwMDAwMDAwWjBYMQswCQYDVQQGEwJQTDEVMBMG
 A1UECgwMb3NzbHNpZ25jb2RlMSAwHgYDVQQLDBdDZXJ0aWZpY2F0aW9uIEF1dGhv
 cml0eTEQMA4GA1UEAwwHUm9vdCBDQTCCASIwDQYJKoZIhvcNAQEBBQADggEPADCC
-AQoCggEBAKzObJwYq4t9Y/OOQLqUNLU8RDXq284L8zQgRLkvApF87FNN7kozIgC9
-/HAgJSho/Lup5lzkCWa3fjkYm1EBrL+JihesSaCxxe7LOg6tRaY+WxikwMUnlkNE
-s3R+DogeGVsla4q0FEcICiz3FHTfSAUVmrN3Nj1ll7npJXrqmXxfCuO3slgjUkHq
-tdZ5t1rSWwbiUhGIQKLzo3/uw2XoOI28qpoOw+0/y8AyjWs8My3u8GrYFr+qh5fx
-Y0Zp0EhhAJo23Xd43XmeVKjuKIOaHu3JiM8sp9K1WFsTvFNAO27TBRn/X0mJCeDX
-T117dQxhWOCcQ/uRGuXICT4ign8MLtUCAwEAAaNjMGEwDwYDVR0TAQH/BAUwAwEB
-/zAdBgNVHQ4EFgQU6ewx3DIpbR8OptEmDFlYNELRqP4wHwYDVR0jBBgwFoAU6ewx
-3DIpbR8OptEmDFlYNELRqP4wDgYDVR0PAQH/BAQDAgGGMA0GCSqGSIb3DQEBCwUA
-A4IBAQAhRMun0IzPmHVFM+SOfSCPVAgogWpqR5XlBAFlS+Aen6v3ukQAQjEhfBbE
-dZG6ye9i0ebf9qXYTvSq5wfaqP7FGb2/Z96uPXNMXPi796KjLW2CG578DitORPb7
-x1eV3UGrQX2bMQ0JbGkBU+DIdIRBqDfad/kjLtm5eHyCbaodSWdaZO4LSUIy3MBx
-2UeBj2qD4RTA0Dt7hG7RA5QdTxHlZyLIk8HX3krZ+il5RmSbOnQs/XqK5DJp4J5p
-122sIO4Y9ki+Wewzx8f3/7mcVbcMo67GwRHo8bk3GjWE74pczyrzfP68vDQ4tn85
-NcLPeLClfSziJD09z+Iyp94EQeKX
+AQoCggEBAKap6aiOWSWzu26AxjlHa4L8sTzYGxQZysvbys3FIxqsztIsWRB0jKq7
+V4x2why9Sgi6tGKWatk0xOckTDiyFFD7BRYf3qDxFivzTkKCvhbvbbxX/6VwcOfR
+XQ0JTQeWFD2o26WWO2MbM0ZXyfZBbiRgIqVi+fbmoEASfYz7eGYET/w7807ERDch
+E3o1XbCgc4mFO/pS3sMykW+2zCz09SGLGKmdkReWXhpiTZ1ZEk9GrqVqKKZ14UIA
+h6nCwoNQpjmQx8KDfjnTYRPYg0N03MjL9i3F11qpYgu6VdZ2l/xaun3NYFBrJ+44
+Pdc+QOqVZZB1YqWqQYPm/OoSlY5p2xECAwEAAaNjMGEwDwYDVR0TAQH/BAUwAwEB
+/zAdBgNVHQ4EFgQUc2QE5V0j8b3ef6gx5kBosb0lE1IwHwYDVR0jBBgwFoAUc2QE
+5V0j8b3ef6gx5kBosb0lE1IwDgYDVR0PAQH/BAQDAgGGMA0GCSqGSIb3DQEBCwUA
+A4IBAQA1jjltAjaXdGmNTLmbD/EvsqCtmX0gS0tXsncbxHEmBNq/gStZg4IhqApw
+LE8uVkjGVxRd3JPsqNL2furjzK1axMZsaNkJD1dWxcT8kfC3+cngTPiWOlCmbjJW
+CjuUWlAUnlFU9VfW0xKMToEC3yU16MSM6L3dIgkp7TC1AfhQLj5Iokn5Cc4r5kzT
+i/sQPlzPYVE+3Onq5VGMMkzK8gmS81Zqndfd0KQvKVdHZrsM88U3+1nKEsQ4dd/3
+8HV+C7LE+ydPV3ywJa6KTtmCMhUlLb2IpNbBgRkro69rkJWtSVDIHjoqwKokGDzx
+kolS+GurfGkHKTbY8HwyfBTQjZR1
 -----END CERTIFICATE-----
--- a/tests/certs/CACertCRL.pem
+++ b/tests/certs/CACertCRL.pem
@@ -2,12 +2,12 @@
 MIIB9zCB4AIBATANBgkqhkiG9w0BAQsFADBgMQswCQYDVQQGEwJQTDEVMBMGA1UE
 CgwMb3NzbHNpZ25jb2RlMSAwHgYDVQQLDBdDZXJ0aWZpY2F0aW9uIEF1dGhvcml0
 eTEYMBYGA1UEAwwPSW50ZXJtZWRpYXRlIENBFw0xOTAxMDEwMDAwMDBaFw00MzAx
-MDEwMDAwMDBaMCcwJQIUOpY5wp7DtqsdxII7sxculedk0PYXDTIyMDcyNjEzMDk1
-NlqgIzAhMB8GA1UdIwQYMBaAFJ4FrAtb4UB/ag702URPiid97ziLMA0GCSqGSIb3
-DQEBCwUAA4IBAQA4Kw0vEJrtjjMam16iN/stOMxJDgkp1IQzA3narxr9fEjX5Ynk
-JztuEExtowPIDOLGWCySXNEMmxCzXNAMvlUq+UQvnWrwgHQ9R7TBYIcAY+VRmzKz
-T4PXvDSL2WMuJ1dLWoIcL2D0wEdti7YMvAnCrOC8HAPGgke5QcOgSfMSAYSAtpiw
-PZAFgcuo53AodlCw9J+CPcHPYw+C2QExOy8s8q6d8Xgjg+Ge7v3RxLWy74sNPl0u
-uZ79vcLNEeqEXxKaw5abqDqIDcUKIT3b62KsSxkak9IGNMLcTASw1V+YaKVLSYNW
-NTuc5WJblfZi/q7WUMKkmRERzvdg2rf0CSH1
+MDEwMDAwMDBaMCcwJQIUDNLHzkIApmatwuk4ju+Cl74nPlsXDTIyMTEyODE0MjIy
+NVqgIzAhMB8GA1UdIwQYMBaAFDOp0wp8AlPBt9gaqapVpq6gXOmcMA0GCSqGSIb3
+DQEBCwUAA4IBAQC2ThTTU7HzwKakJGaeAXsv57I9gQwZJh1sM0CW1bYhQCFQXx9n
+hNnpKu3CezoldjTYgZHphAwrZfLz/u8bKuxuOn/gBT6tRNU7YhFd605Qp6AzCQus
+bENda7LYc+cWbrxtndc54zmEMZ1LkpYE62APN/aCC29BlWQQw+/IvprI10MROlB2
+z3tUjiBNmabHyD2tVLPfuoK3Wdm0upWSYhDJNuFwdIhyZgkYo04YxHrmnfkIp6fD
+HuD341PRMTAItN4334HaJwgu2Oa8MdiiG+ZPlIUSFePgFLgFeNE3EgwTxIu8SQNo
+k5EM7M5+03i7C8p9bBS88n2LzHhoHDI8gfvQ
 -----END X509 CRL-----
--- a/tests/certs/TSA.key
+++ b/tests/certs/TSA.key
@@ -1,28 +1,28 @@
 -----BEGIN PRIVATE KEY-----
-MIIEvAIBADANBgkqhkiG9w0BAQEFAASCBKYwggSiAgEAAoIBAQCivbbTCnbOqPoV
-7VVP/KJzgslx8yfX9laGwTsvzqStQtG8j40ljR85WD/bgy3I5duebudg7JhOVH2f
-aSqbh8SCzP4YNDFcseDIuHdoXi54POgW3S/wbe8l8P7g3btcBgnlXh/izhUlEMib
-Q/8G8UZj0n/MgKMLzcXc4t2eQ4Pzw7xAPqoXBZ20Fg2rFBfUsDDLc5F7lpO5t1WQ
-dXTYmF0oAL/HLmd+HKe5Xgp9jJ6XacesjWLnhRdZ06uUP9cgo+Jem4QZxeFhOtMe
-KAf7JH3Juz2Gi0a+4dMSNzES4m8RAlf6pXh4kAh7EhMNb/Ir6ZpY8uKM7dtn38mh
-1f59EKc9AgMBAAECggEAK5zB0P695hYcpyGqOjxO4LvM9m+eXt7SQ1ynWuF6+j+s
-62ZhAg42rux6eH5IF82ZtHSuJyhgjKVR4RWS6IlS3WbINX5PODMnNUNSJLMQqwJP
-hEkUXs9nRni2JVbmrfukTUaTLvnhasR7rjhjsN2Z6ohv3UMf7rrfapmVoKMhSoLd
-jqvZ2ZoaT5mfTiK/5PDyWqrt/vmvE4VlipAsvudwozG5vQDEsCNDNUTg5OTGnPUU
-i0xeQTioqzCcweIlEaWhj/eMHx/eAeZ6V0Evid/YwTDlbTqVDWZGLJ/phOnKJs3V
-j+eOv7E2d1ga7149SPDIv5Y0YZ91v3M06ICk5o66gQKBgQDSM+YyR5YOJTSAuAcI
-uKTc33wwCbPiJF/F1zLJDdPp2IamZQbuNIX/8fOG3Gho+OnfNAykMcr3rFrug2vi
-9GhWDQqguYGh16xos+2zNan6P/s0/rQ4OfPIsTEGC8X3fJeMzZUNMvnyN/FXzzus
-020o29gu59esEfGHEsvAupC3IQKBgQDGMqGLgrU4oW/5mmm4BZlwwkZJNMmFH49u
-Qe4Ylj87SQduExJMmTfrmANqQXu7RXG0IxLcvhwMLVCCYAkvuBv1awsbg2yfP1Pn
-Wb/K+5CaHaxnpwSpRiGaN6fnAPDl8PnALMVXtQGru2MMcISxOIFQ9slHth0lmaMo
-odIPIL1YnQKBgAOSskUEhn5zD3NorWXujY7blabTY2VirOYWBFz6iTGeZpuJeBaw
-ed6h5DvUn0m5gXAz2EsqNYMEQP9w6HKRKPzdd+LHhHaVze5xsIatUNhaIhECi1mx
-Un2E1Yp+xLyyN3lDPVdeGHWPkeCmOyNy7JYXNpOFiVr5axuarC/4e+FBAoGAeuRR
-/mshaufOwnnYK15tcdlEM4gjnAOhr7/5ng0rT9tMXBg/NHeckNxE4dGQouHASu2k
-eHL4eSRv0ycxCwGhdF7XGEw5QdTGdaDUp0ussaLMj8ijv0HY/AKefUG8HRd6BIq+
-Ik/9pTofhEsQO8LJjCY5T9m/4NyOqlcMJI0sWpECgYBvPLnutbBXYONVAE3jL05K
-hWwenKpv5Aaa11ahqzhil2Tj+VOMtmvhsSc5loSG83qp5LtN4LxyR0Vn9AGN7Z+d
-Ut6LHeZ/DMW3/RPT+1MIKm6WLNxgk5YvuCxprdpfE5tTmV9/t+t3Uao7TRsLPl2o
-qAKz8Fvq0el5RW3EtAgd4A==
+MIIEvwIBADANBgkqhkiG9w0BAQEFAASCBKkwggSlAgEAAoIBAQDngzpJn/umy8rb
+AySHhXZ4jxljzZ/xDW/8txby3fg1LLP1zeP9g5Vwq70B5Z2V1QWevuiCzzEhMKxU
+6JvFUgJnCAqKLxcPxmplgLHIOftq6ly3XicmvwXzD8uE0mDNP50v4hABl66OlRF+
+3cGlwEHlr/A4VYcyEc70xBI2cUjmutr2O2+Ncpamw/iAl60zuKD9CGP7AMfcfUC1
+CdlomyMAk3ROvWoJXxV5jCg7LoTQWSzdzQLi2/mBjG0y9Hl4dlR27jsoeUNlkluS
+C8Dh3mPGQ/rfIEc/en1UO+sVZRo9iBJCw88zjv6uCf2nlevaAUptRuvQqEv4WMNa
+MDe0LqKFAgMBAAECggEADGeQHnTHf5Lozh5XfTlt9Ry5wAq6+Kf/KZvHy4CQNY0t
+IU0EW8eKoMxa5rieXnJZCD3zaFrgp4YbnbsO6Z4wycN5CRln020Os1yXagKNUZe4
+Gw0c7OsMTw84GOhKzoSLkA69S9CMRlB/bPxFzXNFNKDeLq6KfWjN3XiJ/0fHn7Em
+RnHB3C7/z+u70YbkN1T7edzn70NRhBNCVIXuXxrQ946BRJOGyHlarDHQPffGfYzs
+r4Ym7DRd2RyIH3xWdJfEdcQOgioYpQqMiXk+afMNRNbkSpi2ImPACIpX54Zxkjni
+/zjST+CBC/8PZfdmVCW/mRNs2IOmZWCK3yka7JKRgQKBgQD1S9VJbBLHpKAtnahr
+yqwAD82dnaw21BKJaPqeOq4it+EF8p2xmjDCuOXl8qw5t0+SgIalQwWzWVCLoav9
+58ajCJwuTcqnrrmhx0oX3eAYt6z/hnInWdLpDppg8BOxMmFzgPI2+6g3SyXDex29
+RslygQ8/xReUV//LOYayK4GPQQKBgQDxnWt8MnGiQGVaA1x7wM96C0Y8RfuaqfNX
+/eyuw9Z4AB+wg4G2d0NhsbSNdmZ72KuKXaa2Gl3MJ7ImUGbskgTNybQfIwftwcSZ
+Road4j3ZVfNN6QdOEu+c/8xyf4wEf1ygD7qp/geXRSOf1eAC3bg+qAPFRi4Ml9vZ
+WPtQHoeGRQKBgQC1ZeQPzPVCYHL36NxRAmsaDJXA1oYxcblXlqOnebum/xZQgK+v
+VFh016lj2IcWk4Ir1BI+iobfr9AJwoBfXxSdIlPlRi/0uxiaE7euWK1Neh9cQ7+N
+bDa1lNADmebmJly8vwHgOVCXu2Hs4tIGVB/LJrfxL76VPFJ0plD+13+TQQKBgQCu
+tXLMerlaZpAi930/axtbl51w5fXaT3vLq6wiby+80/IWM/FunGyXzuTMycmvUeww
+gCmVUhlzD10tgUP7Wlao62oYfb8098EOSEX9EmRArHrjYlZK6TxkXT9bp8j6U5IJ
+l3YVh7+7bV4mLIPv7p1UiS84zddseb6aOlJ3C6xZtQKBgQCSBAvt4wrMU62+0Zu/
+RAPl0QOR+4WtbVIuoROeYk3Vm5CFooZmE2PcCmatXw2LL9VPQ+wIsc8Ov/c6SNjW
+8FUj2ihdLEhA4pLl7hADAnNBwSnK2nODP1TDoIKG1Hw5yZoYIS4xFz/F96brkaLE
+IiONCCr/bs9tYZNfl92oXvIc/A==
 -----END PRIVATE KEY-----
--- a/tests/certs/TSA.pem
+++ b/tests/certs/TSA.pem
@@ -1,28 +1,28 @@
 -----BEGIN CERTIFICATE-----
-MIIEzTCCA7WgAwIBAgIUfRjXKciCGA4XbhbhxbAwfpcLGmowDQYJKoZIhvcNAQEL
+MIIEzTCCA7WgAwIBAgIUaDV4MOupysdYAfwy1BiaGDN6KWwwDQYJKoZIhvcNAQEL
 BQAwYDELMAkGA1UEBhMCUEwxFTATBgNVBAoMDG9zc2xzaWduY29kZTEkMCIGA1UE
 CwwbVGltZXN0YW1wIEF1dGhvcml0eSBSb290IENBMRQwEgYDVQQDDAtUU0EgUm9v
 dCBDQTAeFw0xODAxMDEwMDAwMDBaFw0yODAxMDEwMDAwMDBaMFUxCzAJBgNVBAYT
 AlBMMRUwEwYDVQQKDAxvc3Nsc2lnbmNvZGUxHDAaBgNVBAsME1RpbWVzdGFtcCBB
 dXRob3JpdHkxETAPBgNVBAMMCFRlc3QgVFNBMIIBIjANBgkqhkiG9w0BAQEFAAOC
-AQ8AMIIBCgKCAQEAor220wp2zqj6Fe1VT/yic4LJcfMn1/ZWhsE7L86krULRvI+N
-JY0fOVg/24MtyOXbnm7nYOyYTlR9n2kqm4fEgsz+GDQxXLHgyLh3aF4ueDzoFt0v
-8G3vJfD+4N27XAYJ5V4f4s4VJRDIm0P/BvFGY9J/zICjC83F3OLdnkOD88O8QD6q
-FwWdtBYNqxQX1LAwy3ORe5aTubdVkHV02JhdKAC/xy5nfhynuV4KfYyel2nHrI1i
-54UXWdOrlD/XIKPiXpuEGcXhYTrTHigH+yR9ybs9hotGvuHTEjcxEuJvEQJX+qV4
-eJAIexITDW/yK+maWPLijO3bZ9/JodX+fRCnPQIDAQABo4IBiDCCAYQwDAYDVR0T
-AQH/BAIwADAWBgNVHSUBAf8EDDAKBggrBgEFBQcDCDAdBgNVHQ4EFgQUKWCqogni
-6SseJ/P6LXo0M2cK++QwHwYDVR0jBBgwFoAU/5nNuG4Tm4v2y9uKf428/4fVQesw
+AQ8AMIIBCgKCAQEA54M6SZ/7psvK2wMkh4V2eI8ZY82f8Q1v/LcW8t34NSyz9c3j
+/YOVcKu9AeWdldUFnr7ogs8xITCsVOibxVICZwgKii8XD8ZqZYCxyDn7aupct14n
+Jr8F8w/LhNJgzT+dL+IQAZeujpURft3BpcBB5a/wOFWHMhHO9MQSNnFI5rra9jtv
+jXKWpsP4gJetM7ig/Qhj+wDH3H1AtQnZaJsjAJN0Tr1qCV8VeYwoOy6E0Fks3c0C
+4tv5gYxtMvR5eHZUdu47KHlDZZJbkgvA4d5jxkP63yBHP3p9VDvrFWUaPYgSQsPP
+M47+rgn9p5Xr2gFKbUbr0KhL+FjDWjA3tC6ihQIDAQABo4IBiDCCAYQwDAYDVR0T
+AQH/BAIwADAWBgNVHSUBAf8EDDAKBggrBgEFBQcDCDAdBgNVHQ4EFgQUVzwbd6xW
+ZJQxaV7Vq7t+rbzjtJQwHwYDVR0jBBgwFoAUZiXazafm/zg+I4wbylR/skX0dWow
 gYQGCCsGAQUFBwEBBHgwdjA5BggrBgEFBQcwAoYtaHR0cDovL1RTQUNBLnRpbWVz
 dGFtcGF1dGhvcml0eS5jb20vVFNBQ0EuY3J0MDkGCCsGAQUFBzABhi1odHRwOi8v
 b2NzcC5UU0FDQS50aW1lc3RhbXBhdXRob3JpdHkuY29tOjkwODAwPgYDVR0fBDcw
 NTAzoDGgL4YtaHR0cDovL1RTQUNBLnRpbWVzdGFtcGF1dGhvcml0eS5jb20vVFNB
 Q0EuY3JsMFUGA1UdHgROMEygGDAKggh0ZXN0LmNvbTAKggh0ZXN0Lm9yZ6EwMAqH
 CAAAAAAAAAAAMCKHIAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAMA0G
-CSqGSIb3DQEBCwUAA4IBAQB4YXa5nVWUzWSsUDMfYFTEETOe8boUErwfrDNBuj6z
-B5en20FhI49i6PCYEfNq3vrAtPOEFJj+KPomN3C46VLxbUEvqWLdq6EyzWvVVmXK
-VLeC0qV0m6CFM8GplaWzZdfFTQaaLUhgY08ZU2gp4QsoS2YjAosxlZrNSm6pBbv3
-q+Og1KeSK8gKS0V89k+6e3LOEF6KaNWKSkoz5xDniQY//mTjiDcNmYUh0KhHfhdU
-eO92M82uJSaDqnRs5HsWPs6z6qdfpuvj++OtQ1VCM2p5SEH2sEomdeN3YYChuG4h
-yzn0mYAdbTyGJHlFm17AH+SQRbVqCKYdHDaqsMb+fWzi
+CSqGSIb3DQEBCwUAA4IBAQBdZ/Zgc0e6F4H7J/rdAegE3MhZPLLAPGq/H5IPC4ij
+ksW6ujTejCJgtSWAfkUXjaCDQvT7bkq4AMo18ELc3I2kpTp1dvftc3aMoDBuGz/7
+K0sxlA1DFncmY6hhh3cg2Gjb1vlnuWnwwSrfy46UuNvsJi4DjBZXHt+gY8KM4Gas
+i3MIqlM/ZIlljc2LRGUKxftkM3odP1kA65yNlcMV9BESRGsAOH8nb6f7Qb4I6vjk
+Oijzs75JPWy9C+vMutH5LSfMAjrdRmNKnDoWVCe22QT1jReIRMNsRjdX0j+SkyyL
+HpqLVmczWLFHPspm8tRyAI7Vc0CpY9ixJyeidD4mg7iv
 -----END CERTIFICATE-----
--- a/tests/certs/TSACA.pem
+++ b/tests/certs/TSACA.pem
@@ -1,22 +1,22 @@
 -----BEGIN CERTIFICATE-----
-MIIDkDCCAnigAwIBAgIUJ0nfE+EVsIThltlY2LHVWMJVIq4wDQYJKoZIhvcNAQEL
+MIIDkDCCAnigAwIBAgIUWSzkBFwFSMGVBrF88OJwBUP9eNkwDQYJKoZIhvcNAQEL
 BQAwYDELMAkGA1UEBhMCUEwxFTATBgNVBAoMDG9zc2xzaWduY29kZTEkMCIGA1UE
 CwwbVGltZXN0YW1wIEF1dGhvcml0eSBSb290IENBMRQwEgYDVQQDDAtUU0EgUm9v
 dCBDQTAeFw0xNzAxMDEwMDAwMDBaFw0yNjExMTAwMDAwMDBaMGAxCzAJBgNVBAYT
 AlBMMRUwEwYDVQQKDAxvc3Nsc2lnbmNvZGUxJDAiBgNVBAsMG1RpbWVzdGFtcCBB
 dXRob3JpdHkgUm9vdCBDQTEUMBIGA1UEAwwLVFNBIFJvb3QgQ0EwggEiMA0GCSqG
-SIb3DQEBAQUAA4IBDwAwggEKAoIBAQDGOTX1f9dmtUiyzlsUInRIGfRMya338SVx
-vYGeOwdpTSSGlYUVwR9AuFewQF5+klelstCJe+SoUG0AdzS30mRWlQrhip4UdvdW
-T2gkNKbSn6DQzlWoQej9izqRLxAsbuszgkvnLOBEmPaLimDsCgu0bAN95Hp0Hls9
-O/fVmzh8VuV4iscxc7q13ZB7CylWgwd55CFEGd/jpJ6kMwSHbOLoBWp4GQ3KxR+c
-ASAo0FapU2WSZB2EYWszRiyq91X+AvIYN4ypTv7RccgfUvnZ2qFykJAkf/wgkynu
-Qg7rCUNfUEpDc7jlqtDWR7iLrtHBkA17C3IU8ymmKQYWfw3ZyBwvAgMBAAGjQjBA
-MA8GA1UdEwEB/wQFMAMBAf8wHQYDVR0OBBYEFP+ZzbhuE5uL9svbin+NvP+H1UHr
-MA4GA1UdDwEB/wQEAwIBBjANBgkqhkiG9w0BAQsFAAOCAQEAbj3aFwIUxvzwgywO
-gj01JM8GNbw1E4MGdkaNI8rgeY8ay15ZXhR9NpRWWb6Y7IXPq5XhuEktVte5Z4Kf
-XLBrr7Xe9VVqJL9zd1tMzOEM/zG77rZf/iXBTZLkCtQc/GOEY4TTWKNEl5hiWVE0
-po97GX5XHoeyHlWQ75sd9z6MxFxmvdp9/uyYD700e9sd5gcD8LGvHw2DNy8vntYV
-ia9h95N9i1umffxU460o8W5GoIcsD13B3YftvnWhGSXqovBRFgcPAQZ4eW9Qh/zA
-4zQBQrRvmREPihXVdgtWVpbRchP99oSZBrYr7Hh/P69rycklquqxJl1ol1wbT6dK
-S5Gmng==
+SIb3DQEBAQUAA4IBDwAwggEKAoIBAQC1iXUDwCHsyISjyimEvnvIPer4aqdpZ26y
+uKI2oURQQpAlL+PlbP5/wm63tyxCY/Oo6cadcXEk9TGEi/SpGMKJtfaZAFzygEQC
+dU995pfpLsQ9Aaf53ErE4L7RsB+noDnYd0UWklBHBmKmWSszfaS/qcsqexP+BBzy
+GgUK1/bVg2SjYESoQ7lWA0o/s9VTYBQ0k36EqykBjWhy87goKBvvDkafqbWRcG+8
+RMgbzvD4vBjOyZWhXH6cKj6Hg7QfDjuTQWV7tQ+zCq0pGMsftcH+N7z5oXyNUiwj
+ALPWxOvrViNRDG2I3GUXU3yWQIHNhmNkIrNE5GJRmx1dkYX0oEW3AgMBAAGjQjBA
+MA8GA1UdEwEB/wQFMAMBAf8wHQYDVR0OBBYEFGYl2s2n5v84PiOMG8pUf7JF9HVq
+MA4GA1UdDwEB/wQEAwIBBjANBgkqhkiG9w0BAQsFAAOCAQEAmCnnUVQoKhbKLqyp
+qNdyT5FFEKyoCd9defa5bKzPhzdkkuxPjidsLsxvP58n6EB8i2inddBfdUxv/0lG
+kGHO+7Lz8qjMlJ9+VlLTmA96Q4SHBP9XKUmyYPx3PpEZcxiJZNXoUO1ZokTps0/Q
+BSoQ+9SPIIzxEzr0BNijtY/x6TJV5M9J1ePWuK1Vu0mhTs3FPE+/wXzxZG94ZQLg
+pX9hkkXmvRIe7dg3XZDUtKglH2p4c22vX7UP4qyqpPWFWKw1lCYt9K7QOVAdEfCI
+TlpZh6MQPZIk0FUAWOxDeRTAJk/Pe0I7wf+L1hwxi1y4JfgebNXynp46T6v/I5Wm
+UApdTw==
 -----END CERTIFICATE-----
--- a/tests/certs/cert.pem
+++ b/tests/certs/cert.pem
@@ -1,46 +1,46 @@
 -----BEGIN CERTIFICATE-----
-MIID7jCCAtagAwIBAgIUdLInHjkevRVCr7I78r5++6eSrZ0wDQYJKoZIhvcNAQEL
+MIID7jCCAtagAwIBAgIUa9RRoV7Ah+0wEQjOlKzlNOaZ+CIwDQYJKoZIhvcNAQEL
 BQAwYDELMAkGA1UEBhMCUEwxFTATBgNVBAoMDG9zc2xzaWduY29kZTEgMB4GA1UE
 CwwXQ2VydGlmaWNhdGlvbiBBdXRob3JpdHkxGDAWBgNVBAMMD0ludGVybWVkaWF0
 ZSBDQTAeFw0xODAxMDEwMDAwMDBaFw0yNDEyMzEwMDAwMDBaMIGdMQswCQYDVQQG
 EwJQTDEZMBcGA1UECAwQTWF6b3ZpYSBQcm92aW5jZTEPMA0GA1UEBwwGV2Fyc2F3
 MRUwEwYDVQQKDAxvc3Nsc2lnbmNvZGUxDDAKBgNVBAsMA0NTUDEUMBIGA1UEAwwL
 Q2VydGlmaWNhdGUxJzAlBgkqhkiG9w0BCQEWGG9zc2xzaWduY29kZUBleGFtcGxl
-LmNvbTCCASIwDQYJKoZIhvcNAQEBBQADggEPADCCAQoCggEBALJLJ3Vty2lgJw+5
-ouAV4ZqkIwvfWPcE7zD1CfQIL2802jVuCSTkN9cfFVYMKFEPJxQWJAKoCzr/Ux8z
-Yt9BXO5o39+z7umLKmc6pfrZJ6kG4msrMjZv36LsCQyfjUc1O9H1aiOQEvRQY2pF
-2v5dfqRMrAqH1ESQHCggUBjElWj9oMFax8jyO7JxTzuttOb6mhDmqz4q2u4LwZGH
-lBofgOAB54Mlv41x7dDh85i/jayXuYYmsjRwCuBAn14+D2zImyPDx5UaUJJMzujo
-QriOZ4KU2dHRgy0+vd7ZbrL1kRY1axyNQ+jBk7UHnlZZ2CCkhBoZIM6ez3ljPwgr
-cpg0RtcCAwEAAaNiMGAwCQYDVR0TBAIwADAdBgNVHQ4EFgQUBxPEs09WXDxGqb+D
-WTFgcUQd0AEwHwYDVR0jBBgwFoAUngWsC1vhQH9qDvTZRE+KJ33vOIswEwYDVR0l
-BAwwCgYIKwYBBQUHAwMwDQYJKoZIhvcNAQELBQADggEBAKK2e1s1puUFbNjglopi
-mKZ4Pks2zb6LVUGG6Q4XQ1dWe25ovt68jWv56HFyCMI1N+L8q0+Ku2eOfLObS7Ej
-FFRUWEIXDgipryDTGzoWRM380fuYpL/j7Rt1/xmIHWTFibf/6gK/naRXsFH3dEbb
-7DDWQ5pAd2d60dB+ThUEIZQTQd/926Kuk5oESvP08fXMYTuiYARypG1gmiuvxQ9N
-mDJP6CHxyJR/LB4tb0RAqnLkVsXVBDnRYWdEvkuhoqTtbhVzVbL3mPeEmVYypxxd
-NdrHpU5zmxFSin2T3F0TneNcT+MDV+dQcWyTGNYs/fnmo85LsiakJixGv1qx8PTs
-8iE=
+LmNvbTCCASIwDQYJKoZIhvcNAQEBBQADggEPADCCAQoCggEBALeV2F3cRYWmG2li
+Bh1qMgaJbK62gE8Cxu/LG6WSFwOutmKSg5BJQDnpkHe8d0a0mpq1mTWlTb2Y+NVh
+vHIMOj/X0EdUg8h7NctODJdb5rHcVQEAs5e8QpDY5cpF3Ec+kDE4meHGvetKzBrv
+PH8lMa/I8HEUnU3qWdTCrMqIvMmjokkaAAHJ22h8uUWnoXhLRh2Az1nqQPrmiVW0
+T8UWFRiWYZr5aFbfrqLamdJxMcS0UnbF6vJzhmpaZVE+7UK4vQYN87MgNbJkAfJQ
+5m5DZWiowQGqiXU3O6Du9EgCRBWv5qlTs0GSdZKh4ziC+ovFT7XQUG9iNoiY+YTV
+b3YGo/ECAwEAAaNiMGAwCQYDVR0TBAIwADAdBgNVHQ4EFgQUA4795c7ElDG1YAeM
+Ei7zxnd53pYwHwYDVR0jBBgwFoAUM6nTCnwCU8G32BqpqlWmrqBc6ZwwEwYDVR0l
+BAwwCgYIKwYBBQUHAwMwDQYJKoZIhvcNAQELBQADggEBAC61GBFi9ao1WEsiRhY4
+MA4Dh5PA8fugwrFmczaDiC34rrVjhCATK8HybLivkLemzN7NxXiNKYjLC7ehYnEt
+dk6vLSUhf1BPxSbzt5aOOYcsATHQlKVYbPs9h2/RIQMCU0w1YEKWsCoDayf92mdG
+aqc1byYDbH2prnwEJsOxgqIH9kBeIOhA91SugoGG5Ppu9J6jCHeuujLXILg3TYA8
+xwlzgIOaQ/pkr65G/7bPUnb+FmZbVZ9fvVzMeqPfGDN3In2VNNt1ukE6syOjrQVQ
+tanpymo51YHCrmFog1731u261eh1k7qlc/gdMtuwcIuS5fgLcw9JuteQ2D2yOMRc
+Hps=
 -----END CERTIFICATE-----
 -----BEGIN CERTIFICATE-----
-MIIDrDCCApSgAwIBAgIUUPCDF21g2spK7557HZUhqSxBltMwDQYJKoZIhvcNAQEL
+MIIDrDCCApSgAwIBAgIUFXuEwy7PJbkt3nd3ZfAc5VsLJKcwDQYJKoZIhvcNAQEL
 BQAwWDELMAkGA1UEBhMCUEwxFTATBgNVBAoMDG9zc2xzaWduY29kZTEgMB4GA1UE
 CwwXQ2VydGlmaWNhdGlvbiBBdXRob3JpdHkxEDAOBgNVBAMMB1Jvb3QgQ0EwHhcN
 MTgwMTAxMDAwMDAwWhcNMjYwMTAxMDAwMDAwWjBgMQswCQYDVQQGEwJQTDEVMBMG
 A1UECgwMb3NzbHNpZ25jb2RlMSAwHgYDVQQLDBdDZXJ0aWZpY2F0aW9uIEF1dGhv
 cml0eTEYMBYGA1UEAwwPSW50ZXJtZWRpYXRlIENBMIIBIjANBgkqhkiG9w0BAQEF
-AAOCAQ8AMIIBCgKCAQEAyMs1XoC0NUT5YgydibOrE5SWBKk5C47B6tv6gA4t3zZJ
-wejaiPkj+aTIU3Ww5DO/Gpz0GuqCHNBczIw92Cfvv8kyWzUy46bRkpBJLFav0JXS
-B3xQaPlHWeXqMfVAGuM5ExT4CjjYKFsrgV1Q300thCHBhvr8TPekDIf+6J7NSz1P
-062pYgypfqsA8OwKaQbgOL9v4QRmHoolnEDc1dK/FS4f3p9dlifl7kcSVGQK0yit
-7Uncn250icCxMxS3MOE2NfuplUOSN6h6poWNGUsx00O7Dy9nUndUwJRpFfKXTV3v
-GtlmFLNoho+ss/usnxjxggWBcRtKhqd8nGSJUlzs0wIDAQABo2YwZDASBgNVHRMB
-Af8ECDAGAQH/AgEAMB0GA1UdDgQWBBSeBawLW+FAf2oO9NlET4onfe84izAfBgNV
-HSMEGDAWgBTp7DHcMiltHw6m0SYMWVg0QtGo/jAOBgNVHQ8BAf8EBAMCAYYwDQYJ
-KoZIhvcNAQELBQADggEBAD/FBa4stJGd/Acg2E2soI071B/l9B7FiqIRpCFuLVC4
-1m7TIcjioIpZrxXwE1Egf8A9/6D/kKZtWnOljcxtPBEb+1/gB61M381RIgoMQ/Pf
-7XX2yakk6mscUjbSTR//Mj1sYOs2r6ueZBp0whzF9nVvA43G6WMpf6XZqmhlg/oV
-ynytW1Iu1SPoru3y8dX/lsukvKCak7MAp1eBcuUJxS56DnKcV9xgC30m3g+CErI3
-qsOJ7lcfDP6fDjy7MfBsZBiY64MqwlDjjn7+Pleo69JedMwurHLhKnfm07DBPy8X
-+EnQk61xHEjQtTsddXyQGQV3yjqylOF2AgsAf256uuA=
+AAOCAQ8AMIIBCgKCAQEA1Jo5DoM/cVNHQ66L+8o4yO20tG3cz5yIJmf157HxpgFA
+UsXOTslEGSt0zM9CbRRuJpV4e6xRQeXJNmoAwT4HyU0vDRE17zPpAkX4bJSM9koj
+dYYjj8ktvu0D9Lou7l3RL4wybqhYO4eFNvDm0CXlAf0GYmuQ+q3o+ykGNMA3P3es
+JpmuMj5fKazV9LUCDjgG9lJxwSb2i/iT3hNTaa/aEvBXoeBqcmRZmjZDD3sUHqEr
+iZFR1v1ezoQKafjKi7eXHoB02Hpy2MnCM6TU8krdpjQhkBC978bZUs6WNkfpfTiT
+6835UsvRWcL0Ea/rGje4Y0FlU0AmAvO5/7j+IqD+NQIDAQABo2YwZDASBgNVHRMB
+Af8ECDAGAQH/AgEAMB0GA1UdDgQWBBQzqdMKfAJTwbfYGqmqVaauoFzpnDAfBgNV
+HSMEGDAWgBRzZATlXSPxvd5/qDHmQGixvSUTUjAOBgNVHQ8BAf8EBAMCAYYwDQYJ
+KoZIhvcNAQELBQADggEBAF4qZoBV9bhp/f86BwaGcuVcY66muoTnFLTgW7TsGDnx
+MKsW5/2CeY4+Yev9LngraEt3DadZJFq+Fl2RzKhRi7KyUmYAzGq3dtGWmYiNgHVH
+gyT1Hl4Kjq9EEsrB8ra+s5/j+zWCkutLn593ROxbj637d7/8BCq73SMBPzQqJVqn
+uluFKOWbDigCGpuEdb8NKDimVQXhg/9ee1tn77btLYkA9+M9fmjoOE0/XVxClYB0
+WulU76/9MdPbgkUnD49AalyBBojQdyj+LqDG3LnYjBBVevuqoF2op8Vna1D5+nVZ
+3+wTPzAR97/mzTDVi1C/R6Dh9gAcKFwhHPrK+R+lWeg=
 -----END CERTIFICATE-----
--- a/tests/certs/crosscert.pem
+++ b/tests/certs/crosscert.pem
@@ -1,23 +1,25 @@
 -----BEGIN CERTIFICATE-----
-MIIDzjCCAragAwIBAgIUGjZdQYlcAtlqZOsQ7eWRimQ9PIcwDQYJKoZIhvcNAQEL
+MIIESDCCAzCgAwIBAgIUFE1jkC6SiKP7a9zwHpiGCY4HBwMwDQYJKoZIhvcNAQEL
 BQAwbzELMAkGA1UEBhMCUEwxFTATBgNVBAoMDG9zc2xzaWduY29kZTEMMAoGA1UE
 CwwDQ1NQMRIwEAYDVQQDDAljcm9zc2NlcnQxJzAlBgkqhkiG9w0BCQEWGG9zc2xz
 aWduY29kZUBleGFtcGxlLmNvbTAeFw0xODAxMDEwMDAwMDBaFw0yMDA2MTkwMDAw
 MDBaMG8xCzAJBgNVBAYTAlBMMRUwEwYDVQQKDAxvc3Nsc2lnbmNvZGUxDDAKBgNV
 BAsMA0NTUDESMBAGA1UEAwwJY3Jvc3NjZXJ0MScwJQYJKoZIhvcNAQkBFhhvc3Ns
 c2lnbmNvZGVAZXhhbXBsZS5jb20wggEiMA0GCSqGSIb3DQEBAQUAA4IBDwAwggEK
-AoIBAQDJU6WNMOEoErLYb5Qc7jsvVgruPM7DJTZ4vUpJNYAyprSDHciaKSa4SiYS
-84Mxc6tzBoJvKOAwpxzzONOqPVWUd5J244urgvfHgSGWsbA8bakiIYlETopnecFk
-B3ZELR33CPqIbpYYMYujhPGFa1xxZxFykJ1iBhZ8Gh3W4wHi/2kW6hTQkihMtUPP
-Xxc2XWACj/tz22OSdgNZcIfhXiy2HOuPch+0UlDR4UmlJIR5aet1y832hHoeeevo
-qfhfGOm9rRf9nyxKDwTyaN7JVOb7A1k6KJEJoe1zfIwT56mgoA433iUWFMLB6hKU
-be3zV1vGjk77Kk7atcvEMTRq+rwHAgMBAAGjYjBgMAkGA1UdEwQCMAAwHQYDVR0O
-BBYEFEXgglEcDh+8oCCvjlxrN/Y7C4YwMB8GA1UdIwQYMBaAFEXgglEcDh+8oCCv
-jlxrN/Y7C4YwMBMGA1UdJQQMMAoGCCsGAQUFBwMDMA0GCSqGSIb3DQEBCwUAA4IB
-AQBo8UqUEjxGQCVU/IgphwKA8Rb/uAyBYm+AjqFDs82lA6ze0n08Bj+eciVkxscA
-0deivOC1sDD88QkLzSQ9CPk4e7+m7nx5SFUnUWY+o3ln+cTbGSM0jW9hme0LtHXX
-QxDSKDBhQonRQk7lQ+TwFR7ol+y5SdZy7YQ+v/25qO6MMQgSPykJIa4vF7lwrYhu
-qL+1MJx/ryTbCUExcKNNkWHZJRc9ZvtdWEHYpBSZl5xmJdKMLnHAu5uv8N2pezzp
-PfujldZky7bnERaTM+bf/LvKXS8RfQGrCLu9QjgPVa6ysZV6gXTsEtwYh64vucjS
-s7IhdLxfiT0xYkK4JWrRLc38
+AoIBAQCr6fvjRqB3eVHQ9YO3r0rgjWCt9pFHpQyXzTgJxIr07aZtOLfJd8hwUHsn
+u8bq5YoSMHElNz7e8aJI2nadgH2JPeOEt+uuEgnBTP4S1Fmbj+ARs8tzpPI8Wjld
+Vp6qU8KXASyjwhWXRUf338xH4aBaxfWqxlKVYuOJX1R9o6KE8AkIhn3UzCf8tkwN
+2r3RqESiawT0XhcKItrXW0ql/Aae8y3N4KSfh0z3pabGz6amSWQIoN/MXM6GElEl
+eQf2PWco2Ik3vCUq8a4JHMpETayjb297AXjX23yMU5wnAunU9obV96st3TwE7Ja6
+GdEbd+ZbtI0uYHIaw9noptyecKd9AgMBAAGjgdswgdgwCQYDVR0TBAIwADAdBgNV
+HQ4EFgQU1wzCqEdaRKbmhVDVvm6N93L+aTEwgZYGA1UdIwSBjjCBi6FzpHEwbzEL
+MAkGA1UEBhMCUEwxFTATBgNVBAoMDG9zc2xzaWduY29kZTEMMAoGA1UECwwDQ1NQ
+MRIwEAYDVQQDDAljcm9zc2NlcnQxJzAlBgkqhkiG9w0BCQEWGG9zc2xzaWduY29k
+ZUBleGFtcGxlLmNvbYIUFE1jkC6SiKP7a9zwHpiGCY4HBwMwEwYDVR0lBAwwCgYI
+KwYBBQUHAwMwDQYJKoZIhvcNAQELBQADggEBAC7N6qRKLabyA4xrC/fp6AQOe+zr
+lFORWevFhSNmmM02wtl4XK22jx4nxFp136ZMw1z2Hm5ljJd6vjr5QMeT7tPvs8Zi
+ivWC6zl2z1oqbNiU+mGO366ZDJUL9dhZqPTFydocL1cj0IBlOQemQwQpnaeTBp8Y
+d8AJqlgZ0d0Eocnv2C4MFoig8Y8/nDIROQ7nhL6gPxTApYoRcZw//XbKnoGZGPsU
+V9MJjfL1QW28KCpZyIxdc5f985kHuxZ6EHSXI3hSHArILRCFLIgOcoOA/kh1u4sI
+QyFpcqOHwj+3pAwcdyZsfxGKGC1M5qyX3o3nb2VoeEF13MPGf5t6nX0iTJc=
 -----END CERTIFICATE-----
--- a/tests/certs/expired.pem
+++ b/tests/certs/expired.pem
@@ -1,45 +1,45 @@
 -----BEGIN CERTIFICATE-----
-MIID6jCCAtKgAwIBAgIUdtBZJAw8/6JVNMiQpN3PEROI8rowDQYJKoZIhvcNAQEL
+MIID6jCCAtKgAwIBAgIUC/qlHkKhzuSWn/K7W/WNcYKde1UwDQYJKoZIhvcNAQEL
 BQAwYDELMAkGA1UEBhMCUEwxFTATBgNVBAoMDG9zc2xzaWduY29kZTEgMB4GA1UE
 CwwXQ2VydGlmaWNhdGlvbiBBdXRob3JpdHkxGDAWBgNVBAMMD0ludGVybWVkaWF0
 ZSBDQTAeFw0xODAxMDEwMDAwMDBaFw0xOTAxMDEwMDAwMDBaMIGZMQswCQYDVQQG
 EwJQTDEZMBcGA1UECAwQTWF6b3ZpYSBQcm92aW5jZTEPMA0GA1UEBwwGV2Fyc2F3
 MRUwEwYDVQQKDAxvc3Nsc2lnbmNvZGUxDDAKBgNVBAsMA0NTUDEQMA4GA1UEAwwH
 RXhwaXJlZDEnMCUGCSqGSIb3DQEJARYYb3NzbHNpZ25jb2RlQGV4YW1wbGUuY29t
-MIIBIjANBgkqhkiG9w0BAQEFAAOCAQ8AMIIBCgKCAQEAsksndW3LaWAnD7mi4BXh
-mqQjC99Y9wTvMPUJ9AgvbzTaNW4JJOQ31x8VVgwoUQ8nFBYkAqgLOv9THzNi30Fc
-7mjf37Pu6YsqZzql+tknqQbiaysyNm/fouwJDJ+NRzU70fVqI5AS9FBjakXa/l1+
-pEysCofURJAcKCBQGMSVaP2gwVrHyPI7snFPO6205vqaEOarPira7gvBkYeUGh+A
-4AHngyW/jXHt0OHzmL+NrJe5hiayNHAK4ECfXj4PbMibI8PHlRpQkkzO6OhCuI5n
-gpTZ0dGDLT693tlusvWRFjVrHI1D6MGTtQeeVlnYIKSEGhkgzp7PeWM/CCtymDRG
-1wIDAQABo2IwYDAJBgNVHRMEAjAAMB0GA1UdDgQWBBQHE8SzT1ZcPEapv4NZMWBx
-RB3QATAfBgNVHSMEGDAWgBSeBawLW+FAf2oO9NlET4onfe84izATBgNVHSUEDDAK
-BggrBgEFBQcDAzANBgkqhkiG9w0BAQsFAAOCAQEAX1Ar7jRAXdcA0Wu37yRi58QN
-hpa1VLXadqfB+i5Y4e3DzqnMbpkLWsFzreC1AG0RjLe52s4PRUE6boGlpUeAyfFC
-Qu2Gl/REVWwMCYV8bq3vQZkYQjklAXCQLWFk5TrzuDmBcV8+fY518nWw+xmcYwW5
-5oehLsvB4nxoBzlHgcdDwS5b2dmpCKCbZFLU9aA9DjAVvY/9B8emyj7Sh2sEK0Yf
-xwHlATTVq5O0/9tvVZQmYsbpS0iCRGBM+spTEhDT4WGsaRO6wP+Ucgp6Ym3ahMvz
-tHME3uUanKWVoDb69sguGZ6KlnZZZdIX1AJ3dlTXCrzEO9xsoAzqzsxVJGrraQ==
+MIIBIjANBgkqhkiG9w0BAQEFAAOCAQ8AMIIBCgKCAQEAt5XYXdxFhaYbaWIGHWoy
+BolsrraATwLG78sbpZIXA662YpKDkElAOemQd7x3RrSamrWZNaVNvZj41WG8cgw6
+P9fQR1SDyHs1y04Ml1vmsdxVAQCzl7xCkNjlykXcRz6QMTiZ4ca960rMGu88fyUx
+r8jwcRSdTepZ1MKsyoi8yaOiSRoAAcnbaHy5RaeheEtGHYDPWepA+uaJVbRPxRYV
+GJZhmvloVt+uotqZ0nExxLRSdsXq8nOGalplUT7tQri9Bg3zsyA1smQB8lDmbkNl
+aKjBAaqJdTc7oO70SAJEFa/mqVOzQZJ1kqHjOIL6i8VPtdBQb2I2iJj5hNVvdgaj
+8QIDAQABo2IwYDAJBgNVHRMEAjAAMB0GA1UdDgQWBBQDjv3lzsSUMbVgB4wSLvPG
+d3neljAfBgNVHSMEGDAWgBQzqdMKfAJTwbfYGqmqVaauoFzpnDATBgNVHSUEDDAK
+BggrBgEFBQcDAzANBgkqhkiG9w0BAQsFAAOCAQEAxFOT4ZK5vzlGSUAd+KNsYbsf
+ws5xIroPVIkxSkF5Z99IMOVxlg0Q87u2WWXqdNfXeb973PJL+iWVtIooKUOCLJSY
+r1cVkJ7rV+Y0LROkpIRpFa4x3oNTkO4g26rsn6u9iJmMUXrgbvNbuuwIwH1pZWnV
+7wYLmKBk/gmhJAi1S0wZVfGs/BcPylUt9z1OUBd32HGzXKUSEkd/YAlJr2PdVSI0
+7HZ30qjzxoiPlR4mAr4skr2aI8cioRnXAuppa147U+yVTTxKa4AJkhj2ujbQovo/
+ERyf8dTrWbYZ4GywB1A/I08Lf24w+COWabCTsCioDBFCO1YLKS5QuETtqKxf/Q==
 -----END CERTIFICATE-----
 -----BEGIN CERTIFICATE-----
-MIIDrDCCApSgAwIBAgIUUPCDF21g2spK7557HZUhqSxBltMwDQYJKoZIhvcNAQEL
+MIIDrDCCApSgAwIBAgIUFXuEwy7PJbkt3nd3ZfAc5VsLJKcwDQYJKoZIhvcNAQEL
 BQAwWDELMAkGA1UEBhMCUEwxFTATBgNVBAoMDG9zc2xzaWduY29kZTEgMB4GA1UE
 CwwXQ2VydGlmaWNhdGlvbiBBdXRob3JpdHkxEDAOBgNVBAMMB1Jvb3QgQ0EwHhcN
 MTgwMTAxMDAwMDAwWhcNMjYwMTAxMDAwMDAwWjBgMQswCQYDVQQGEwJQTDEVMBMG
 A1UECgwMb3NzbHNpZ25jb2RlMSAwHgYDVQQLDBdDZXJ0aWZpY2F0aW9uIEF1dGhv
 cml0eTEYMBYGA1UEAwwPSW50ZXJtZWRpYXRlIENBMIIBIjANBgkqhkiG9w0BAQEF
-AAOCAQ8AMIIBCgKCAQEAyMs1XoC0NUT5YgydibOrE5SWBKk5C47B6tv6gA4t3zZJ
-wejaiPkj+aTIU3Ww5DO/Gpz0GuqCHNBczIw92Cfvv8kyWzUy46bRkpBJLFav0JXS
-B3xQaPlHWeXqMfVAGuM5ExT4CjjYKFsrgV1Q300thCHBhvr8TPekDIf+6J7NSz1P
-062pYgypfqsA8OwKaQbgOL9v4QRmHoolnEDc1dK/FS4f3p9dlifl7kcSVGQK0yit
-7Uncn250icCxMxS3MOE2NfuplUOSN6h6poWNGUsx00O7Dy9nUndUwJRpFfKXTV3v
-GtlmFLNoho+ss/usnxjxggWBcRtKhqd8nGSJUlzs0wIDAQABo2YwZDASBgNVHRMB
-Af8ECDAGAQH/AgEAMB0GA1UdDgQWBBSeBawLW+FAf2oO9NlET4onfe84izAfBgNV
-HSMEGDAWgBTp7DHcMiltHw6m0SYMWVg0QtGo/jAOBgNVHQ8BAf8EBAMCAYYwDQYJ
-KoZIhvcNAQELBQADggEBAD/FBa4stJGd/Acg2E2soI071B/l9B7FiqIRpCFuLVC4
-1m7TIcjioIpZrxXwE1Egf8A9/6D/kKZtWnOljcxtPBEb+1/gB61M381RIgoMQ/Pf
-7XX2yakk6mscUjbSTR//Mj1sYOs2r6ueZBp0whzF9nVvA43G6WMpf6XZqmhlg/oV
-ynytW1Iu1SPoru3y8dX/lsukvKCak7MAp1eBcuUJxS56DnKcV9xgC30m3g+CErI3
-qsOJ7lcfDP6fDjy7MfBsZBiY64MqwlDjjn7+Pleo69JedMwurHLhKnfm07DBPy8X
-+EnQk61xHEjQtTsddXyQGQV3yjqylOF2AgsAf256uuA=
+AAOCAQ8AMIIBCgKCAQEA1Jo5DoM/cVNHQ66L+8o4yO20tG3cz5yIJmf157HxpgFA
+UsXOTslEGSt0zM9CbRRuJpV4e6xRQeXJNmoAwT4HyU0vDRE17zPpAkX4bJSM9koj
+dYYjj8ktvu0D9Lou7l3RL4wybqhYO4eFNvDm0CXlAf0GYmuQ+q3o+ykGNMA3P3es
+JpmuMj5fKazV9LUCDjgG9lJxwSb2i/iT3hNTaa/aEvBXoeBqcmRZmjZDD3sUHqEr
+iZFR1v1ezoQKafjKi7eXHoB02Hpy2MnCM6TU8krdpjQhkBC978bZUs6WNkfpfTiT
+6835UsvRWcL0Ea/rGje4Y0FlU0AmAvO5/7j+IqD+NQIDAQABo2YwZDASBgNVHRMB
+Af8ECDAGAQH/AgEAMB0GA1UdDgQWBBQzqdMKfAJTwbfYGqmqVaauoFzpnDAfBgNV
+HSMEGDAWgBRzZATlXSPxvd5/qDHmQGixvSUTUjAOBgNVHQ8BAf8EBAMCAYYwDQYJ
+KoZIhvcNAQELBQADggEBAF4qZoBV9bhp/f86BwaGcuVcY66muoTnFLTgW7TsGDnx
+MKsW5/2CeY4+Yev9LngraEt3DadZJFq+Fl2RzKhRi7KyUmYAzGq3dtGWmYiNgHVH
+gyT1Hl4Kjq9EEsrB8ra+s5/j+zWCkutLn593ROxbj637d7/8BCq73SMBPzQqJVqn
+uluFKOWbDigCGpuEdb8NKDimVQXhg/9ee1tn77btLYkA9+M9fmjoOE0/XVxClYB0
+WulU76/9MdPbgkUnD49AalyBBojQdyj+LqDG3LnYjBBVevuqoF2op8Vna1D5+nVZ
+3+wTPzAR97/mzTDVi1C/R6Dh9gAcKFwhHPrK+R+lWeg=
 -----END CERTIFICATE-----
--- a/tests/certs/intermediate.pem
+++ b/tests/certs/intermediate.pem
@@ -1,22 +1,22 @@
 -----BEGIN CERTIFICATE-----
-MIIDrDCCApSgAwIBAgIUUPCDF21g2spK7557HZUhqSxBltMwDQYJKoZIhvcNAQEL
+MIIDrDCCApSgAwIBAgIUFXuEwy7PJbkt3nd3ZfAc5VsLJKcwDQYJKoZIhvcNAQEL
 BQAwWDELMAkGA1UEBhMCUEwxFTATBgNVBAoMDG9zc2xzaWduY29kZTEgMB4GA1UE
 CwwXQ2VydGlmaWNhdGlvbiBBdXRob3JpdHkxEDAOBgNVBAMMB1Jvb3QgQ0EwHhcN
 MTgwMTAxMDAwMDAwWhcNMjYwMTAxMDAwMDAwWjBgMQswCQYDVQQGEwJQTDEVMBMG
 A1UECgwMb3NzbHNpZ25jb2RlMSAwHgYDVQQLDBdDZXJ0aWZpY2F0aW9uIEF1dGhv
 cml0eTEYMBYGA1UEAwwPSW50ZXJtZWRpYXRlIENBMIIBIjANBgkqhkiG9w0BAQEF
-AAOCAQ8AMIIBCgKCAQEAyMs1XoC0NUT5YgydibOrE5SWBKk5C47B6tv6gA4t3zZJ
-wejaiPkj+aTIU3Ww5DO/Gpz0GuqCHNBczIw92Cfvv8kyWzUy46bRkpBJLFav0JXS
-B3xQaPlHWeXqMfVAGuM5ExT4CjjYKFsrgV1Q300thCHBhvr8TPekDIf+6J7NSz1P
-062pYgypfqsA8OwKaQbgOL9v4QRmHoolnEDc1dK/FS4f3p9dlifl7kcSVGQK0yit
-7Uncn250icCxMxS3MOE2NfuplUOSN6h6poWNGUsx00O7Dy9nUndUwJRpFfKXTV3v
-GtlmFLNoho+ss/usnxjxggWBcRtKhqd8nGSJUlzs0wIDAQABo2YwZDASBgNVHRMB
-Af8ECDAGAQH/AgEAMB0GA1UdDgQWBBSeBawLW+FAf2oO9NlET4onfe84izAfBgNV
-HSMEGDAWgBTp7DHcMiltHw6m0SYMWVg0QtGo/jAOBgNVHQ8BAf8EBAMCAYYwDQYJ
-KoZIhvcNAQELBQADggEBAD/FBa4stJGd/Acg2E2soI071B/l9B7FiqIRpCFuLVC4
-1m7TIcjioIpZrxXwE1Egf8A9/6D/kKZtWnOljcxtPBEb+1/gB61M381RIgoMQ/Pf
-7XX2yakk6mscUjbSTR//Mj1sYOs2r6ueZBp0whzF9nVvA43G6WMpf6XZqmhlg/oV
-ynytW1Iu1SPoru3y8dX/lsukvKCak7MAp1eBcuUJxS56DnKcV9xgC30m3g+CErI3
-qsOJ7lcfDP6fDjy7MfBsZBiY64MqwlDjjn7+Pleo69JedMwurHLhKnfm07DBPy8X
-+EnQk61xHEjQtTsddXyQGQV3yjqylOF2AgsAf256uuA=
+AAOCAQ8AMIIBCgKCAQEA1Jo5DoM/cVNHQ66L+8o4yO20tG3cz5yIJmf157HxpgFA
+UsXOTslEGSt0zM9CbRRuJpV4e6xRQeXJNmoAwT4HyU0vDRE17zPpAkX4bJSM9koj
+dYYjj8ktvu0D9Lou7l3RL4wybqhYO4eFNvDm0CXlAf0GYmuQ+q3o+ykGNMA3P3es
+JpmuMj5fKazV9LUCDjgG9lJxwSb2i/iT3hNTaa/aEvBXoeBqcmRZmjZDD3sUHqEr
+iZFR1v1ezoQKafjKi7eXHoB02Hpy2MnCM6TU8krdpjQhkBC978bZUs6WNkfpfTiT
+6835UsvRWcL0Ea/rGje4Y0FlU0AmAvO5/7j+IqD+NQIDAQABo2YwZDASBgNVHRMB
+Af8ECDAGAQH/AgEAMB0GA1UdDgQWBBQzqdMKfAJTwbfYGqmqVaauoFzpnDAfBgNV
+HSMEGDAWgBRzZATlXSPxvd5/qDHmQGixvSUTUjAOBgNVHQ8BAf8EBAMCAYYwDQYJ
+KoZIhvcNAQELBQADggEBAF4qZoBV9bhp/f86BwaGcuVcY66muoTnFLTgW7TsGDnx
+MKsW5/2CeY4+Yev9LngraEt3DadZJFq+Fl2RzKhRi7KyUmYAzGq3dtGWmYiNgHVH
+gyT1Hl4Kjq9EEsrB8ra+s5/j+zWCkutLn593ROxbj637d7/8BCq73SMBPzQqJVqn
+uluFKOWbDigCGpuEdb8NKDimVQXhg/9ee1tn77btLYkA9+M9fmjoOE0/XVxClYB0
+WulU76/9MdPbgkUnD49AalyBBojQdyj+LqDG3LnYjBBVevuqoF2op8Vna1D5+nVZ
+3+wTPzAR97/mzTDVi1C/R6Dh9gAcKFwhHPrK+R+lWeg=
 -----END CERTIFICATE-----
--- a/tests/certs/key.pem
+++ b/tests/certs/key.pem
@@ -1,27 +1,28 @@
------BEGIN RSA PRIVATE KEY-----
-MIIEogIBAAKCAQEAsksndW3LaWAnD7mi4BXhmqQjC99Y9wTvMPUJ9AgvbzTaNW4J
-JOQ31x8VVgwoUQ8nFBYkAqgLOv9THzNi30Fc7mjf37Pu6YsqZzql+tknqQbiaysy
-Nm/fouwJDJ+NRzU70fVqI5AS9FBjakXa/l1+pEysCofURJAcKCBQGMSVaP2gwVrH
-yPI7snFPO6205vqaEOarPira7gvBkYeUGh+A4AHngyW/jXHt0OHzmL+NrJe5hiay
-NHAK4ECfXj4PbMibI8PHlRpQkkzO6OhCuI5ngpTZ0dGDLT693tlusvWRFjVrHI1D
-6MGTtQeeVlnYIKSEGhkgzp7PeWM/CCtymDRG1wIDAQABAoIBACR/jgxT9ZgUvupR
-Li6BTDXD9AiyKBwpPm3fO7JhGpTBVQorBGQw891t14hN5NLzLyTFg4mnrOTe770r
-X8okL0n+3hWFWBsnCf8n2mKHob7QUfluVlEehcFsYE3dO6agFybb/mZZUAgDjNZs
-hnAb45juuSlOtP10Is90DfGEDLH5IeY1xjzc7Qv/CFxCffIAC8QmQYUTihJ2m5aE
-7Hvs79oEcSvbRJDYbykzrJ0eeIaEvfOxkWJnnJHrhiONzb8qgj3DLiZdX1qeo1Ao
-ldNxEG9n3Axd6M0nhajz0qbDV11S8YiKfP10XRQh5xv7lZi7MjvrRxFTFYpSrXwi
-YYyFNoECgYEA6YnIYg1nIe3qaI3Me1RQQTGRl8M/dQc2d98Pz5mez9vN3TIW6nEs
-QYw/9OKG1ercbD+YnuaV+1izuAcA3mNlSDReTtzInAotJjTH1V3WYqvBTVUZSb6T
-5qSBfRDC3AULFvHeX5c5wqgfB3U9KLDfVBfaHnMerg6dAGsYZPhPMrkCgYEAw3Eg
-5BRTzbqG0WXF25rycTeHCcylMZRjI+TVcIa8AGqNSCK4HgAWp89XPIV3WceVqe2R
-Lyn1jtA2MgGGcMBDFlOWF+h9j2/j27Z+pyIbBF9LAraJuBOG6dezd50y5Ur6HK/f
-e5lnjvElIYdz+RX+rmw8NFcIUbSAfE7rGinDvg8CgYBDx86VzsgJC/FFySn4/X6R
-fV6BSpTHVYGUhvQiz7ZNI8F7GoeWIaSznY4OeBSkT5cL/+U+8TPEkHkQx0+UPArw
-Suq4PtImn7l85kK9hY+scacX18QQKDTq8wH2F4BGtVwDm81rtwt3mK3wzzEh9zvK
-P2X6AnV8FReyQGMDIyJxWQKBgD6nu9WitqMTkzj6GY4nhGXLWV1I4ASe/5F4QPzM
-FOVFQ3nGt6PWf2zYyay7VOHRXCeX451tJC3ejiFF3+WxnVBBB7Muc2JSiofbX4sR
-Ifwq2I9MGaaLjArXfc9w8+oSOVCNCWZEbbCjmjW/iOxnorgkNsDIzf/zj+VKH5DJ
-ptj9AoGAUpB6nPES3Nnj8DsbpHCnShYMl/rFxzxg2pJJosXzuS/ttuBT2DlT5eiY
-aZcL1DGSp2CD4QXbVuDHPkQMPzVfZzKAuCZlotEMR9byK46aYIkQueym45e0PGZP
-djKZm+cxF+W55jBkBBhV6wSOLNRWCUYiIIq3RmwWjkopNvSlwY4=
------END RSA PRIVATE KEY-----
+-----BEGIN PRIVATE KEY-----
+MIIEvAIBADANBgkqhkiG9w0BAQEFAASCBKYwggSiAgEAAoIBAQC3ldhd3EWFphtp
+YgYdajIGiWyutoBPAsbvyxulkhcDrrZikoOQSUA56ZB3vHdGtJqatZk1pU29mPjV
+YbxyDDo/19BHVIPIezXLTgyXW+ax3FUBALOXvEKQ2OXKRdxHPpAxOJnhxr3rSswa
+7zx/JTGvyPBxFJ1N6lnUwqzKiLzJo6JJGgABydtofLlFp6F4S0YdgM9Z6kD65olV
+tE/FFhUYlmGa+WhW366i2pnScTHEtFJ2xeryc4ZqWmVRPu1CuL0GDfOzIDWyZAHy
+UOZuQ2VoqMEBqol1Nzug7vRIAkQVr+apU7NBknWSoeM4gvqLxU+10FBvYjaImPmE
+1W92BqPxAgMBAAECggEABgXOcVXWmgnXAGAM304qkfhULm8iSnVG3XsGTyo5ww8q
+QJWZ51Pf3YaCDMuMBi/KmqS2gAIKSNwJ3cY1DnDEQG/uD0kAgcU5AJBw7U5AhTk7
+OKszORkd/e5/q2EbVuMAi/zVX4jLljp/86pBi9JGjQPEj5Ejj/jONaKlVIej2eGM
+EqD5vmTP4TyCVybByXnYve7NxqhBvrgGgfoyGZLckohUPCBNc7Xr69M+X07Hmxtw
+uyx7PqSJEBm22bBzAjqZYAgLU6rIZ/HNdhkue/r1MlTQvaYFZsEiMUgm46G5yxXf
+3LKAMl0LJRg4lO6bljfKf883sOB4ukHGvOydOS3I2QKBgQDBk9Dl72SUspeISTMi
+v7ceNpDKpKnIygfXMeC5BJTEPu8DsQiyAQltlfeAi4yNALHyWlGso0Cr8kbt49Wc
+puGT8fmcJ4Y0LJTrhoYhZvdpKfthubLTbvZ0BXEk9dRvO24vdidiK3mzYgvIUBBy
+TW1t97gg1lGzugghJxWV9VhFLwKBgQDyySoZbrpCaxK59PErGzZNje1u+3/ER4U/
+WzqxsG9wHl5nn5OmYN2uJf2/Mcy4i3IVBimn6NMoBt4CkmgKg9XHXbvUmlpp+rXq
+YIXgwKLEIOiKkTlQAiQXGLY6gnK45MY/HAfGYt48Gczqban0+ZhVDLbYqm3opNTV
+qcsGTCug3wKBgEy4PtTCmkRsu63iBQhCn1q0GZKWzEut9omNVRk+/Y8BukzmPyH9
+UzVBeIRTynKf0ebRcQlQ8nKzZPUnlSOu6DxwbJM8/u1ytdrayBI48tmVL/I9XOh7
+U+YLuhoB7B3hmfvpghZgUK2NsEnFoZ8LnNE2mp05E8x9cHdjPCTgxgk1AoGAVr4G
+u5VzGL3Wr/fSnpsXWLgnp6pVqNQAyHUY0FmARyeihRVflGcxbMIIGeJ6CUlRFi/g
+fCxUHQ3vDi4u2+rkW1mi/T+0o/6+NiIMUxOgU8CKNM/UEUDikhnzKY5n3KVoKy4h
+nQC/8+sn6m4J9gujIGftgBXMNP2HSVtjvTlDsTMCgYANc6p+FuMMDsXfcV19iAQB
+Vtioc7xi2HlvqgrOLIqJa+P7CQjYbF382l9ncpG7LmkPVf1D0CK/unCz13P4ZR0L
+voYHOtkWDb33+L61v0IWw1FC9hE1rFe//LRKFCkjcILpAdRhumNpPHak0/wNTqjB
+MKUIYvx+w976iv/A5rSEXA==
+-----END PRIVATE KEY-----
--- a/tests/certs/keyp.pem
+++ b/tests/certs/keyp.pem
@@ -1,30 +1,30 @@
------BEGIN RSA PRIVATE KEY-----
-Proc-Type: 4,ENCRYPTED
-DEK-Info: DES-EDE3-CBC,6DB255395263EDD5
-
-A83CH4mv7u89RbT8Tpz4gxImKGBw69Zr3LstRXlliGYobby5YBx/YimkA9wFCZK/
-B1eitWwy0F6axDZQYHKi+yvUJDPB9arXtClvYknMhwrHV1GHLSYeqb9oZJeqNyDZ
-Qd3lxTbUtkMVJgJNrl2kOezm6/srnAw4uA7NyAvrs2vEzB48q4VlysrJq+f9mLXj
-vCmrfUwY999lSifRmoxqeWlqNUQ2tgHHXYMNagpvxYXsfe1UvEH9aa6+UsO9S0py
-7dyfSu95QpWyVNqkWi1VAtNbo2VpjJ6NJLAk+dy/rNqN3a4KnWeIzkYssWTsKB08
-VyHrRLePPx5qdarsczZtf+M2PxhHHnl/09+Nrp5BUcMV31j3v3VSM0K+CHEucAk7
-95rUtSUNywKUQeSgXrG1eLX7kXwRQ7PPCz3sPtvmJRIvQlGgLiTbKPsG52m3kAIw
-zgXPcH2lYb3855occznl72jMUucxXq8gq0bC4xbEg/yJV8p1IuUEFhLGI9s9T4JV
-cO9NKwmSjpHFo1ULhB7o6uMmV2rYDK+5GbQHxZgHj3ES+i53eFMWGubPEEOmqSh3
-9K7gtW9y971tNfhp9ba8RnYXT6xW2nMTM4koO9b4ptdwRo5bMKFvWY8eecXfsDAG
-OJ+aXkDr8jsn1Xauq1m4TM71wn2wUx95KaCpL55UNBEn7aH9qlNFfxdyzXMPYS36
-zgWK44BV3PTSIGgW61NVwHwzi3bFfymortVxGpelzy2dggWVvI3uzKocLFQ8f4oe
-Y0HWSmGVPF3uFHNCZtCB/Tpbz6YwP/YYStqAcryeS5Yo9Hdkh8pBVnYiKdTFEUW2
-RbClgeB2MV+zttsqvmodfkviS4BjWgoIV5szxWOePnO8kQAHA/Ml3CyDPOX6rqI5
-lDKiUojEMLgir/3YWWcmigEIgRsyF3CL1s+kTR3S8e9QRe8RiliRUKW5gXrLEa6j
-eUjs4kgCrvmQvwyZYJjWl+r7ycmk+yB/EZs8P39KRR/pfeZDUCZIOx8vkJBt57hC
-oTNJ5llFzRcmEraElXmDOAuvmj3Lx/4qzY545rtzll3mFHJEX4qITslIX1ksZz1p
-DncuqgIECzmZIeHPbnw7Nkv6EkoPzTOlccqnCH/SumFr2fhctv9x13gGcO4kSsqO
-63yZCFHjMz/mos3l51aIAizj5wQO3BOo+RyKoSQohvPzVtSgjhYMZsAPzXKxF0/H
-9VH0DekEb1WwPSbGZw8kpx9ePlglGqOBinTL6QW8YmFPbjy9RDd1di+fxh4Qe024
-8gERZpdSsMoZ+NP0nr/TSbDISFPqcLzzMpC+V8Fc/QkNvSkR1GLlNQrxLoyfK0VX
-0evysXDxqIWK9+TH7hIS1lf3i2gMkLMppMso1v2Cqf0zRj1oM3MI743QE9XTXKRz
-iAwaEDDPZWS/00T9fqNrHgtSPNpsbeYZQPYaC2lq1kTIEOlUfZZvMy5lxVPVZ8y2
-foit+0DewZsqLDJwbjZ3wYMERVEY7KagoInQa3A1ZC9SkFiCb4fNEbRF13gfjrSz
-muRbKAhEhkzJDFRocIaTKZPIWdvC73tAW66v1Zha74mxuckgnQPPqQ==
------END RSA PRIVATE KEY-----
+-----BEGIN ENCRYPTED PRIVATE KEY-----
+MIIFHDBOBgkqhkiG9w0BBQ0wQTApBgkqhkiG9w0BBQwwHAQIOTZv8WJMOWACAggA
+MAwGCCqGSIb3DQIJBQAwFAYIKoZIhvcNAwcECJGTN+jUwUhuBIIEyJghDPvGvTtV
+0zIVbjh6xm5Uwp9KuhnKq87SW1C8cvL3sQFxcYalUkR5etejHDsrgw6r/jpijh+T
+mRW2GncLHUJjzS/nM3j8UbbY5ELAhcJSQSw5UdtUVJ3kQxL/f4GIlbhx711s4ARV
+YH4BZuOxC0eaobMSOzsYOAY2JPHdRWqhr63acsnn+1nJAhlH7Sz8B7Y5tVNl99O+
+QIsKi6K4s0ySZlU34DMTI78oM49fsi1sAGIR+klbyNvkyqsVmy8YlKLZPpCUzaJd
+a4KZl4rDJtLv4AMXJYJpxOAnP/WDWrnyfguJB9uHie6bgwp/cHUqNGeFBHNgxv5a
+e1/Hk7iNPS9CW23MrFWLOEwB6yv4QuQQg1MlbVsmAT+cc7dEI1h7g6aWzGrZoW+l
+xaElJ9j/0+/8BzXUYy1Uq5Gs2ZwpY6nRHBI4gPOLN8J1piIhPzURIQoPlmsr/47v
+O+duE9zqPZjqEEZMrE00XpJvPmnAOc5eSHCGkf7xiCaqxsVKgCHNGMF+EtYNbOh4
+e/riE7mhhV7SCmJNcnTFTjP6pWskb/HEOyAVbxuN4BQ6GIQTUcbOT9Yy/Azzzs03
+4s1+qYqlzM2RV9WbLd0U21K6emE5JvR80Gis6xol+BiTyjchhGjhuPjyDDPIqQKm
+JCh44UdI7dm4xGd473SbVDxkRRf5WB0VC2BsZMb+RLAaZWsxF1WbGAUsF3coo8Q1
+dnjCjerK6CD7SDSMRyQLvd0SyyDY13GKQkSjGsghl3rEB4FhuyhNPxGxSxrVSkCX
+FvsIiJhc80VBdkJmSg+l1UTisrrBgpGQ+pA8cOtSAwtm+notUvj84iRpAVxN/cuv
+jCL5BrCNuCtrhJEjs24gnMJHEk7FYCp4iPXKh1dveL+NyEDVVR8gJiKddWmdcJ/G
+Cez40UIxuTgHyCDRa2oN8JxSzVt5/QU34WIP+HnZ61rqmZ7nm8CJxp5Hp6Z4bX1J
+QujW3E3FHJ7BtM4wZR7MumKfZsxswvdXU6X1owXiZ4wiL7dqt5jA82LrL7cTq25l
+vg17mZf4Gtwhrbl+/RnTpzPg/qUI1bw2S+Cbtd39Gy6KH/CvVov+E1dQdsFXBATz
+9Cb8JtulYSz6mcfEQUGcz82lP9m6pd2bEYvnljkzd4D6AqtI9bqjuKnOfTlR+E8r
+5trNm3i/sexs3YNWcNJY7IbJ8Lg3LS3ByJG3g9nxueU0tAzzxcRC2VwcjIci1cyY
+lY71JWiEoBsjNR8AlbVvWDtRCo2bu8q3wFLKZMlMZlZG95CqcqXy3QWby1AN4sVv
+tfFldCNVCFLXBVo2lMvw2+J8D129t5bUchd96dGHkQ0aPSnVdMg1AK3g1XxLWc17
+yafjWcR/Pua8zMosN1YqZSDgav7lzEuFDs94cukduop+98T/TjlxejoOimppCtXs
+bKQ+tDUbCg6HRn6vMyWMonrVTaT+RFXtoTfwFEONfjaJ7Wgh2yG/xB6ZjOpMF5Mq
+SbbmF1fh+xJfb6J68RT9mhY9PwOZXKzfcQ4FWonvo14JbQIve8xlJ+gHBUEwBZli
+hkBmZAd0ZGoszPbArcgn3EoIoogG1LcHjWaq+Dw2HYBCtvxZf4e4npg59IG12WOP
+cpkk1XK/52QEeuTSf9x4OA==
+-----END ENCRYPTED PRIVATE KEY-----
--- a/tests/certs/revoked.pem
+++ b/tests/certs/revoked.pem
@@ -1,45 +1,45 @@
 -----BEGIN CERTIFICATE-----
-MIIDvTCCAqWgAwIBAgIUOpY5wp7DtqsdxII7sxculedk0PYwDQYJKoZIhvcNAQEL
+MIIDvTCCAqWgAwIBAgIUDNLHzkIApmatwuk4ju+Cl74nPlswDQYJKoZIhvcNAQEL
 BQAwYDELMAkGA1UEBhMCUEwxFTATBgNVBAoMDG9zc2xzaWduY29kZTEgMB4GA1UE
 CwwXQ2VydGlmaWNhdGlvbiBBdXRob3JpdHkxGDAWBgNVBAMMD0ludGVybWVkaWF0
 ZSBDQTAeFw0xODAxMDEwMDAwMDBaFw0yNDEyMzEwMDAwMDBaMG0xCzAJBgNVBAYT
 AlBMMRUwEwYDVQQKDAxvc3Nsc2lnbmNvZGUxDDAKBgNVBAsMA0NTUDEQMA4GA1UE
 AwwHUmV2b2tlZDEnMCUGCSqGSIb3DQEJARYYb3NzbHNpZ25jb2RlQGV4YW1wbGUu
-Y29tMIIBIjANBgkqhkiG9w0BAQEFAAOCAQ8AMIIBCgKCAQEAsksndW3LaWAnD7mi
-4BXhmqQjC99Y9wTvMPUJ9AgvbzTaNW4JJOQ31x8VVgwoUQ8nFBYkAqgLOv9THzNi
-30Fc7mjf37Pu6YsqZzql+tknqQbiaysyNm/fouwJDJ+NRzU70fVqI5AS9FBjakXa
-/l1+pEysCofURJAcKCBQGMSVaP2gwVrHyPI7snFPO6205vqaEOarPira7gvBkYeU
-Gh+A4AHngyW/jXHt0OHzmL+NrJe5hiayNHAK4ECfXj4PbMibI8PHlRpQkkzO6OhC
-uI5ngpTZ0dGDLT693tlusvWRFjVrHI1D6MGTtQeeVlnYIKSEGhkgzp7PeWM/CCty
-mDRG1wIDAQABo2IwYDAJBgNVHRMEAjAAMB0GA1UdDgQWBBQHE8SzT1ZcPEapv4NZ
-MWBxRB3QATAfBgNVHSMEGDAWgBSeBawLW+FAf2oO9NlET4onfe84izATBgNVHSUE
-DDAKBggrBgEFBQcDAzANBgkqhkiG9w0BAQsFAAOCAQEAYVJiPrkACW/tK487fYS/
-gYzU3fYVCTfHpAv3njarNzy8UBNqBYr0kDg0DLoOWqGV7ogTtlbQP4IIjAQI/kW6
-cEreW8yU5VxO+kxDo+7oG9VEbR85i6kQW2ubJsXV6yBtf5aAbXEqImYrtjh7UObb
-BbQiUI1ll2dXWqvZGxr3Fz1uz8nPMYlBpVjpCh6JF8otdWwABmxRnqUvoLO6BZbH
-/gdUkouXfio9BlWkWaJXJGXMW8B7ozpjuCHSHyfvGKDA3YIfa7++A1BIKxW72jIF
-jRJDw/rwnV59tiEcBWmp2T6vV+rD8yaS+LotRPYD/ck/jEj/mV+N077KLmuZpdJF
-ag==
+Y29tMIIBIjANBgkqhkiG9w0BAQEFAAOCAQ8AMIIBCgKCAQEAt5XYXdxFhaYbaWIG
+HWoyBolsrraATwLG78sbpZIXA662YpKDkElAOemQd7x3RrSamrWZNaVNvZj41WG8
+cgw6P9fQR1SDyHs1y04Ml1vmsdxVAQCzl7xCkNjlykXcRz6QMTiZ4ca960rMGu88
+fyUxr8jwcRSdTepZ1MKsyoi8yaOiSRoAAcnbaHy5RaeheEtGHYDPWepA+uaJVbRP
+xRYVGJZhmvloVt+uotqZ0nExxLRSdsXq8nOGalplUT7tQri9Bg3zsyA1smQB8lDm
+bkNlaKjBAaqJdTc7oO70SAJEFa/mqVOzQZJ1kqHjOIL6i8VPtdBQb2I2iJj5hNVv
+dgaj8QIDAQABo2IwYDAJBgNVHRMEAjAAMB0GA1UdDgQWBBQDjv3lzsSUMbVgB4wS
+LvPGd3neljAfBgNVHSMEGDAWgBQzqdMKfAJTwbfYGqmqVaauoFzpnDATBgNVHSUE
+DDAKBggrBgEFBQcDAzANBgkqhkiG9w0BAQsFAAOCAQEAcB1h5O55YyIuLSPwBKy3
+L36Msm0EutKj07soEVzWyq1/2elOvIksS/O/25zzenR/0Sf1/CBBU1kSQlb0PnkR
+f0cgWKW3I9eqJevkyABE7CS5V8y9YYkrWl+XedRmPBuMYElTs79LQllFPhbga4lQ
+bPtyCM+BpnsK5dkVf0AOoXkMOg7Ib3yg5LAvU2caZ3l7MnX/aXJEeVnJ4tGF6nzI
+1e9X+ypOJ1GOpA/hXEHiDri5FVEMZamVFDRprdB2vA1Go7utNHZ/lvcoHRsU3rR9
+K/uN2Hw4XhFZoN+9F8N0wD8CXk/hjJwPTAcUfzgSjhymgHzZEJa4G40O4fN4jWTr
+Cg==
 -----END CERTIFICATE-----
 -----BEGIN CERTIFICATE-----
-MIIDrDCCApSgAwIBAgIUUPCDF21g2spK7557HZUhqSxBltMwDQYJKoZIhvcNAQEL
+MIIDrDCCApSgAwIBAgIUFXuEwy7PJbkt3nd3ZfAc5VsLJKcwDQYJKoZIhvcNAQEL
 BQAwWDELMAkGA1UEBhMCUEwxFTATBgNVBAoMDG9zc2xzaWduY29kZTEgMB4GA1UE
 CwwXQ2VydGlmaWNhdGlvbiBBdXRob3JpdHkxEDAOBgNVBAMMB1Jvb3QgQ0EwHhcN
 MTgwMTAxMDAwMDAwWhcNMjYwMTAxMDAwMDAwWjBgMQswCQYDVQQGEwJQTDEVMBMG
 A1UECgwMb3NzbHNpZ25jb2RlMSAwHgYDVQQLDBdDZXJ0aWZpY2F0aW9uIEF1dGhv
 cml0eTEYMBYGA1UEAwwPSW50ZXJtZWRpYXRlIENBMIIBIjANBgkqhkiG9w0BAQEF
-AAOCAQ8AMIIBCgKCAQEAyMs1XoC0NUT5YgydibOrE5SWBKk5C47B6tv6gA4t3zZJ
-wejaiPkj+aTIU3Ww5DO/Gpz0GuqCHNBczIw92Cfvv8kyWzUy46bRkpBJLFav0JXS
-B3xQaPlHWeXqMfVAGuM5ExT4CjjYKFsrgV1Q300thCHBhvr8TPekDIf+6J7NSz1P
-062pYgypfqsA8OwKaQbgOL9v4QRmHoolnEDc1dK/FS4f3p9dlifl7kcSVGQK0yit
-7Uncn250icCxMxS3MOE2NfuplUOSN6h6poWNGUsx00O7Dy9nUndUwJRpFfKXTV3v
-GtlmFLNoho+ss/usnxjxggWBcRtKhqd8nGSJUlzs0wIDAQABo2YwZDASBgNVHRMB
-Af8ECDAGAQH/AgEAMB0GA1UdDgQWBBSeBawLW+FAf2oO9NlET4onfe84izAfBgNV
-HSMEGDAWgBTp7DHcMiltHw6m0SYMWVg0QtGo/jAOBgNVHQ8BAf8EBAMCAYYwDQYJ
-KoZIhvcNAQELBQADggEBAD/FBa4stJGd/Acg2E2soI071B/l9B7FiqIRpCFuLVC4
-1m7TIcjioIpZrxXwE1Egf8A9/6D/kKZtWnOljcxtPBEb+1/gB61M381RIgoMQ/Pf
-7XX2yakk6mscUjbSTR//Mj1sYOs2r6ueZBp0whzF9nVvA43G6WMpf6XZqmhlg/oV
-ynytW1Iu1SPoru3y8dX/lsukvKCak7MAp1eBcuUJxS56DnKcV9xgC30m3g+CErI3
-qsOJ7lcfDP6fDjy7MfBsZBiY64MqwlDjjn7+Pleo69JedMwurHLhKnfm07DBPy8X
-+EnQk61xHEjQtTsddXyQGQV3yjqylOF2AgsAf256uuA=
+AAOCAQ8AMIIBCgKCAQEA1Jo5DoM/cVNHQ66L+8o4yO20tG3cz5yIJmf157HxpgFA
+UsXOTslEGSt0zM9CbRRuJpV4e6xRQeXJNmoAwT4HyU0vDRE17zPpAkX4bJSM9koj
+dYYjj8ktvu0D9Lou7l3RL4wybqhYO4eFNvDm0CXlAf0GYmuQ+q3o+ykGNMA3P3es
+JpmuMj5fKazV9LUCDjgG9lJxwSb2i/iT3hNTaa/aEvBXoeBqcmRZmjZDD3sUHqEr
+iZFR1v1ezoQKafjKi7eXHoB02Hpy2MnCM6TU8krdpjQhkBC978bZUs6WNkfpfTiT
+6835UsvRWcL0Ea/rGje4Y0FlU0AmAvO5/7j+IqD+NQIDAQABo2YwZDASBgNVHRMB
+Af8ECDAGAQH/AgEAMB0GA1UdDgQWBBQzqdMKfAJTwbfYGqmqVaauoFzpnDAfBgNV
+HSMEGDAWgBRzZATlXSPxvd5/qDHmQGixvSUTUjAOBgNVHQ8BAf8EBAMCAYYwDQYJ
+KoZIhvcNAQELBQADggEBAF4qZoBV9bhp/f86BwaGcuVcY66muoTnFLTgW7TsGDnx
+MKsW5/2CeY4+Yev9LngraEt3DadZJFq+Fl2RzKhRi7KyUmYAzGq3dtGWmYiNgHVH
+gyT1Hl4Kjq9EEsrB8ra+s5/j+zWCkutLn593ROxbj637d7/8BCq73SMBPzQqJVqn
+uluFKOWbDigCGpuEdb8NKDimVQXhg/9ee1tn77btLYkA9+M9fmjoOE0/XVxClYB0
+WulU76/9MdPbgkUnD49AalyBBojQdyj+LqDG3LnYjBBVevuqoF2op8Vna1D5+nVZ
+3+wTPzAR97/mzTDVi1C/R6Dh9gAcKFwhHPrK+R+lWeg=
 -----END CERTIFICATE-----
--- a/tests/certs/tsa-chain.pem
+++ b/tests/certs/tsa-chain.pem
@@ -1,50 +1,50 @@
 -----BEGIN CERTIFICATE-----
-MIIEzTCCA7WgAwIBAgIUfRjXKciCGA4XbhbhxbAwfpcLGmowDQYJKoZIhvcNAQEL
+MIIEzTCCA7WgAwIBAgIUaDV4MOupysdYAfwy1BiaGDN6KWwwDQYJKoZIhvcNAQEL
 BQAwYDELMAkGA1UEBhMCUEwxFTATBgNVBAoMDG9zc2xzaWduY29kZTEkMCIGA1UE
 CwwbVGltZXN0YW1wIEF1dGhvcml0eSBSb290IENBMRQwEgYDVQQDDAtUU0EgUm9v
 dCBDQTAeFw0xODAxMDEwMDAwMDBaFw0yODAxMDEwMDAwMDBaMFUxCzAJBgNVBAYT
 AlBMMRUwEwYDVQQKDAxvc3Nsc2lnbmNvZGUxHDAaBgNVBAsME1RpbWVzdGFtcCBB
 dXRob3JpdHkxETAPBgNVBAMMCFRlc3QgVFNBMIIBIjANBgkqhkiG9w0BAQEFAAOC
-AQ8AMIIBCgKCAQEAor220wp2zqj6Fe1VT/yic4LJcfMn1/ZWhsE7L86krULRvI+N
-JY0fOVg/24MtyOXbnm7nYOyYTlR9n2kqm4fEgsz+GDQxXLHgyLh3aF4ueDzoFt0v
-8G3vJfD+4N27XAYJ5V4f4s4VJRDIm0P/BvFGY9J/zICjC83F3OLdnkOD88O8QD6q
-FwWdtBYNqxQX1LAwy3ORe5aTubdVkHV02JhdKAC/xy5nfhynuV4KfYyel2nHrI1i
-54UXWdOrlD/XIKPiXpuEGcXhYTrTHigH+yR9ybs9hotGvuHTEjcxEuJvEQJX+qV4
-eJAIexITDW/yK+maWPLijO3bZ9/JodX+fRCnPQIDAQABo4IBiDCCAYQwDAYDVR0T
-AQH/BAIwADAWBgNVHSUBAf8EDDAKBggrBgEFBQcDCDAdBgNVHQ4EFgQUKWCqogni
-6SseJ/P6LXo0M2cK++QwHwYDVR0jBBgwFoAU/5nNuG4Tm4v2y9uKf428/4fVQesw
+AQ8AMIIBCgKCAQEA54M6SZ/7psvK2wMkh4V2eI8ZY82f8Q1v/LcW8t34NSyz9c3j
+/YOVcKu9AeWdldUFnr7ogs8xITCsVOibxVICZwgKii8XD8ZqZYCxyDn7aupct14n
+Jr8F8w/LhNJgzT+dL+IQAZeujpURft3BpcBB5a/wOFWHMhHO9MQSNnFI5rra9jtv
+jXKWpsP4gJetM7ig/Qhj+wDH3H1AtQnZaJsjAJN0Tr1qCV8VeYwoOy6E0Fks3c0C
+4tv5gYxtMvR5eHZUdu47KHlDZZJbkgvA4d5jxkP63yBHP3p9VDvrFWUaPYgSQsPP
+M47+rgn9p5Xr2gFKbUbr0KhL+FjDWjA3tC6ihQIDAQABo4IBiDCCAYQwDAYDVR0T
+AQH/BAIwADAWBgNVHSUBAf8EDDAKBggrBgEFBQcDCDAdBgNVHQ4EFgQUVzwbd6xW
+ZJQxaV7Vq7t+rbzjtJQwHwYDVR0jBBgwFoAUZiXazafm/zg+I4wbylR/skX0dWow
 gYQGCCsGAQUFBwEBBHgwdjA5BggrBgEFBQcwAoYtaHR0cDovL1RTQUNBLnRpbWVz
 dGFtcGF1dGhvcml0eS5jb20vVFNBQ0EuY3J0MDkGCCsGAQUFBzABhi1odHRwOi8v
 b2NzcC5UU0FDQS50aW1lc3RhbXBhdXRob3JpdHkuY29tOjkwODAwPgYDVR0fBDcw
 NTAzoDGgL4YtaHR0cDovL1RTQUNBLnRpbWVzdGFtcGF1dGhvcml0eS5jb20vVFNB
 Q0EuY3JsMFUGA1UdHgROMEygGDAKggh0ZXN0LmNvbTAKggh0ZXN0Lm9yZ6EwMAqH
 CAAAAAAAAAAAMCKHIAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAMA0G
-CSqGSIb3DQEBCwUAA4IBAQB4YXa5nVWUzWSsUDMfYFTEETOe8boUErwfrDNBuj6z
-B5en20FhI49i6PCYEfNq3vrAtPOEFJj+KPomN3C46VLxbUEvqWLdq6EyzWvVVmXK
-VLeC0qV0m6CFM8GplaWzZdfFTQaaLUhgY08ZU2gp4QsoS2YjAosxlZrNSm6pBbv3
-q+Og1KeSK8gKS0V89k+6e3LOEF6KaNWKSkoz5xDniQY//mTjiDcNmYUh0KhHfhdU
-eO92M82uJSaDqnRs5HsWPs6z6qdfpuvj++OtQ1VCM2p5SEH2sEomdeN3YYChuG4h
-yzn0mYAdbTyGJHlFm17AH+SQRbVqCKYdHDaqsMb+fWzi
+CSqGSIb3DQEBCwUAA4IBAQBdZ/Zgc0e6F4H7J/rdAegE3MhZPLLAPGq/H5IPC4ij
+ksW6ujTejCJgtSWAfkUXjaCDQvT7bkq4AMo18ELc3I2kpTp1dvftc3aMoDBuGz/7
+K0sxlA1DFncmY6hhh3cg2Gjb1vlnuWnwwSrfy46UuNvsJi4DjBZXHt+gY8KM4Gas
+i3MIqlM/ZIlljc2LRGUKxftkM3odP1kA65yNlcMV9BESRGsAOH8nb6f7Qb4I6vjk
+Oijzs75JPWy9C+vMutH5LSfMAjrdRmNKnDoWVCe22QT1jReIRMNsRjdX0j+SkyyL
+HpqLVmczWLFHPspm8tRyAI7Vc0CpY9ixJyeidD4mg7iv
 -----END CERTIFICATE-----
 -----BEGIN CERTIFICATE-----
-MIIDkDCCAnigAwIBAgIUJ0nfE+EVsIThltlY2LHVWMJVIq4wDQYJKoZIhvcNAQEL
+MIIDkDCCAnigAwIBAgIUWSzkBFwFSMGVBrF88OJwBUP9eNkwDQYJKoZIhvcNAQEL
 BQAwYDELMAkGA1UEBhMCUEwxFTATBgNVBAoMDG9zc2xzaWduY29kZTEkMCIGA1UE
 CwwbVGltZXN0YW1wIEF1dGhvcml0eSBSb290IENBMRQwEgYDVQQDDAtUU0EgUm9v
 dCBDQTAeFw0xNzAxMDEwMDAwMDBaFw0yNjExMTAwMDAwMDBaMGAxCzAJBgNVBAYT
 AlBMMRUwEwYDVQQKDAxvc3Nsc2lnbmNvZGUxJDAiBgNVBAsMG1RpbWVzdGFtcCBB
 dXRob3JpdHkgUm9vdCBDQTEUMBIGA1UEAwwLVFNBIFJvb3QgQ0EwggEiMA0GCSqG
-SIb3DQEBAQUAA4IBDwAwggEKAoIBAQDGOTX1f9dmtUiyzlsUInRIGfRMya338SVx
-vYGeOwdpTSSGlYUVwR9AuFewQF5+klelstCJe+SoUG0AdzS30mRWlQrhip4UdvdW
-T2gkNKbSn6DQzlWoQej9izqRLxAsbuszgkvnLOBEmPaLimDsCgu0bAN95Hp0Hls9
-O/fVmzh8VuV4iscxc7q13ZB7CylWgwd55CFEGd/jpJ6kMwSHbOLoBWp4GQ3KxR+c
-ASAo0FapU2WSZB2EYWszRiyq91X+AvIYN4ypTv7RccgfUvnZ2qFykJAkf/wgkynu
-Qg7rCUNfUEpDc7jlqtDWR7iLrtHBkA17C3IU8ymmKQYWfw3ZyBwvAgMBAAGjQjBA
-MA8GA1UdEwEB/wQFMAMBAf8wHQYDVR0OBBYEFP+ZzbhuE5uL9svbin+NvP+H1UHr
-MA4GA1UdDwEB/wQEAwIBBjANBgkqhkiG9w0BAQsFAAOCAQEAbj3aFwIUxvzwgywO
-gj01JM8GNbw1E4MGdkaNI8rgeY8ay15ZXhR9NpRWWb6Y7IXPq5XhuEktVte5Z4Kf
-XLBrr7Xe9VVqJL9zd1tMzOEM/zG77rZf/iXBTZLkCtQc/GOEY4TTWKNEl5hiWVE0
-po97GX5XHoeyHlWQ75sd9z6MxFxmvdp9/uyYD700e9sd5gcD8LGvHw2DNy8vntYV
-ia9h95N9i1umffxU460o8W5GoIcsD13B3YftvnWhGSXqovBRFgcPAQZ4eW9Qh/zA
-4zQBQrRvmREPihXVdgtWVpbRchP99oSZBrYr7Hh/P69rycklquqxJl1ol1wbT6dK
-S5Gmng==
+SIb3DQEBAQUAA4IBDwAwggEKAoIBAQC1iXUDwCHsyISjyimEvnvIPer4aqdpZ26y
+uKI2oURQQpAlL+PlbP5/wm63tyxCY/Oo6cadcXEk9TGEi/SpGMKJtfaZAFzygEQC
+dU995pfpLsQ9Aaf53ErE4L7RsB+noDnYd0UWklBHBmKmWSszfaS/qcsqexP+BBzy
+GgUK1/bVg2SjYESoQ7lWA0o/s9VTYBQ0k36EqykBjWhy87goKBvvDkafqbWRcG+8
+RMgbzvD4vBjOyZWhXH6cKj6Hg7QfDjuTQWV7tQ+zCq0pGMsftcH+N7z5oXyNUiwj
+ALPWxOvrViNRDG2I3GUXU3yWQIHNhmNkIrNE5GJRmx1dkYX0oEW3AgMBAAGjQjBA
+MA8GA1UdEwEB/wQFMAMBAf8wHQYDVR0OBBYEFGYl2s2n5v84PiOMG8pUf7JF9HVq
+MA4GA1UdDwEB/wQEAwIBBjANBgkqhkiG9w0BAQsFAAOCAQEAmCnnUVQoKhbKLqyp
+qNdyT5FFEKyoCd9defa5bKzPhzdkkuxPjidsLsxvP58n6EB8i2inddBfdUxv/0lG
+kGHO+7Lz8qjMlJ9+VlLTmA96Q4SHBP9XKUmyYPx3PpEZcxiJZNXoUO1ZokTps0/Q
+BSoQ+9SPIIzxEzr0BNijtY/x6TJV5M9J1ePWuK1Vu0mhTs3FPE+/wXzxZG94ZQLg
+pX9hkkXmvRIe7dg3XZDUtKglH2p4c22vX7UP4qyqpPWFWKw1lCYt9K7QOVAdEfCI
+TlpZh6MQPZIk0FUAWOxDeRTAJk/Pe0I7wf+L1hwxi1y4JfgebNXynp46T6v/I5Wm
+UApdTw==
 -----END CERTIFICATE-----
--- /dev/null
+++ b/tests/certs/tsa-serial
@@ -0,0 +1 @@
+ba458168626824fa059d68686df0f1eb
--- a/tests/conf/makecerts.sh
+++ b/tests/conf/makecerts.sh
@@ -1,4 +1,4 @@
-#!/bin/sh
+#!/bin/bash
 
 result=0
 
@@ -18,7 +18,6 @@
   script_path=$(pwd)
   cd "${result_path}"
   mkdir "tmp/"
-  rm -rf "../certs"
 
 # OpenSSL settings
   CONF="${script_path}/openssl_intermediate.cnf"
@@ -34,7 +33,7 @@
   touch "CA/index.txt"
   echo -n "unique_subject = no" > "CA/index.txt.attr"
   $OPENSSL rand -hex 16 > "CA/serial"
-  $OPENSSL rand -hex 16 > "tsa-serial"
+  $OPENSSL rand -hex 16 > "tmp/tsa-serial"
   echo 1001 > "CA/crlnumber"
   date > "makecerts.log"
   "$OPENSSL" version 2>> "makecerts.log" 1>&2
@@ -175,7 +174,25 @@
       2>> "makecerts.log" 1>&2
   test_result $?
 
-  printf "\nConvert the certificate and the key into a PKCS#12 container\n" >> "makecerts.log"
+  ssl_version=$("$OPENSSL" version)
+  if test "${ssl_version:8:1}" -eq 3
+  then
+    printf "\nConvert the certificate and the key into legacy PKCS#12 container with\
+ RC2-40-CBC private key and certificate encryption algorithm\n" >> "makecerts.log"
+    "$OPENSSL" pkcs12 -export -in tmp/cert.pem -inkey tmp/key.pem -out tmp/legacy.p12 -passout pass:"$password" \
+        -keypbe rc2-40-cbc -certpbe rc2-40-cbc -legacy \
+        2>> "makecerts.log" 1>&2
+  else
+    printf "\nConvert the certificate and the key into legacy PKCS#12 container with\
+ RC2-40-CBC private key and certificate encryption algorithm\n" >> "makecerts.log"
+    "$OPENSSL" pkcs12 -export -in tmp/cert.pem -inkey tmp/key.pem -out tmp/legacy.p12 -passout pass:"$password" \
+        -keypbe rc2-40-cbc -certpbe rc2-40-cbc \
+        2>> "makecerts.log" 1>&2
+  fi
+  test_result $?
+
+  printf "\nConvert the certificate and the key into a PKCS#12 container with\
+  AES-256-CBC private key and certificate encryption algorithm\n" >> "makecerts.log"
   "$OPENSSL" pkcs12 -export -in tmp/cert.pem -inkey tmp/key.pem -out tmp/cert.p12 -passout pass:"$password" \
       -keypbe aes-256-cbc -certpbe aes-256-cbc \
       2>> "makecerts.log" 1>&2
@@ -230,9 +247,9 @@
       -a -s tmp/key.pem -a -s tmp/keyp.pem -a -s tmp/key.der -a -s tmp/key.pvk \
       -a -s tmp/cert.pem -a -s tmp/cert.p12 -a -s tmp/cert.der -a -s tmp/cert.spc \
       -a -s tmp/crosscert.pem -a -s tmp/expired.pem -a -s tmp/revoked.pem -a -s tmp/revoked.spc \
-      -a -s tmp/TSA.pem -a -s tmp/TSA.key -a -s tmp/tsa-chain.pem
+      -a -s tmp/TSA.pem -a -s tmp/TSA.key -a -s tmp/tsa-chain.pem -a -s tmp/legacy.p12
   then
-    mkdir "../certs"
+    mkdir -p "../certs"
     cp tmp/* ../certs
     printf "%s" "keys & certificates successfully generated"
   else
--- a/tests/conf/openssl_tsa_root.cnf
+++ b/tests/conf/openssl_tsa_root.cnf
@@ -27,6 +27,7 @@
 default_startdate               = 20180101000000Z
 default_enddate                 = 20280101000000Z
 unique_subject                  = no
+email_in_dn                     = no
 x509_extensions                 = tsa_extensions
 
 [ policy_match ]
